From 16464d476923e478a00e96f9b419e428f9e5d3b0 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Tue, 19 Dec 2023 18:00:54 +0100
Subject: [PATCH 6/6] Remove declaration of libc functions

---
 w3mbookmark.c    |    1 -
 w3mhelperpanel.c |    1 -
 2 files changed, 2 deletions(-)

--- a/w3mbookmark.c
+++ b/w3mbookmark.c
@@ -183,7 +183,6 @@ insert_bookmark(char *bmark, struct pars
 int
 main(void)
 {
-    extern char *getenv(const char *);
     char *p;
     int length;
     Str qs = NULL;
--- a/w3mhelperpanel.c
+++ b/w3mhelperpanel.c
@@ -167,7 +167,6 @@ int
 main(void)
 {
     Str mailcapfile;
-    extern char *getenv(const char *);
     char *p;
     int length;
     Str qs = NULL;
