From 4fc565cb40416e4752b0e99de3b1de348f2adf42 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Sun, 17 Apr 2022 15:59:41 +0200
Subject: [PATCH 010/172] Link to IRC in the README

---
 README | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/README b/README
index f418b2ac..39785765 100644
--- a/README
+++ b/README
@@ -11,3 +11,5 @@ version <https://sourceforge.net/projects/w3m/>.
 If you can read English, see doc/*.
 If you can read Japanese, see doc-jp/*.
 If you can read both, read both and correct English. :-)
+
+Questions? Join https://web.libera.chat/#w3m !
-- 
2.30.2

