From c515ea8a47b62408943390f579b2f377464f6658 Mon Sep 17 00:00:00 2001
From: Tatsuya Kinoshita <tats@debian.org>
Date: Fri, 29 Apr 2022 10:25:54 +0900
Subject: [PATCH 011/172] Update ChangeLog

---
 ChangeLog | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/ChangeLog b/ChangeLog
index 50ca38f9..93242db3 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,10 @@
+2022-04-29  Tatsuya Kinoshita  <tats@debian.org>
+
+	* NEWS: Update NEWS for w3m 0.5.3+git20220429.
+
+	* config.guess, config.sub:
+	Update config.* with autotools-dev 20220109.1.
+
 2022-04-25  NRK  <nrk@disroot.org>
 
 	Cppcheck fixes
-- 
2.30.2

