From 6bb3b3a8227a1a89f391734d4a4b1d34b226030b Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 21:38:44 +0100
Subject: [PATCH 020/172] Remove unused function

---
 indep.c | 18 ------------------
 indep.h |  1 -
 2 files changed, 19 deletions(-)

--- a/indep.c
+++ b/indep.c
@@ -341,24 +341,6 @@ strcasematch(char *s1, char *s2)
     return (*s2 == '\0');
 }
 
-/* search multiple strings */
-int
-strcasemstr(char *str, char *srch[], char **ret_ptr)
-{
-    int i;
-    while (*str) {
-	for (i = 0; srch[i]; i++) {
-	    if (strcasematch(str, srch[i])) {
-		if (ret_ptr)
-		    *ret_ptr = str;
-		return i;
-	    }
-	}
-	str++;
-    }
-    return -1;
-}
-
 int
 strmatchlen(const char *s1, const char *s2, int maxlen)
 {
--- a/indep.h
+++ b/indep.h
@@ -59,7 +59,6 @@ extern int strncasecmp(const char *s1, c
 #ifndef HAVE_STRCASESTR
 extern char *strcasestr(const char *s1, const char *s2);
 #endif
-extern int strcasemstr(char *str, char *srch[], char **ret_ptr);
 int strmatchlen(const char *s1, const char *s2, int maxlen);
 extern char *remove_space(char *str);
 extern int non_null(char *s);
