From ca0413a84533e774f38c46b3684534f8959c65b4 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 21:41:46 +0100
Subject: [PATCH 023/172] Remove unused function

---
 istream.c | 20 --------------------
 1 file changed, 20 deletions(-)

--- a/istream.c
+++ b/istream.c
@@ -276,26 +276,6 @@ ISgets_to_growbuf(InputStream stream, st
     return;
 }
 
-#ifdef unused
-int
-ISread(InputStream stream, Str buf, int count)
-{
-    int len;
-
-    if (count + 1 > buf->area_size) {
-	char *newptr = GC_MALLOC_ATOMIC(count + 1);
-	memcpy(newptr, buf->ptr, buf->length);
-	newptr[buf->length] = '\0';
-	buf->ptr = newptr;
-	buf->area_size = count + 1;
-    }
-    len = ISread_n(stream, buf->ptr, count);
-    buf->length = (len > 0) ? len : 0;
-    buf->ptr[buf->length] = '\0';
-    return (len > 0) ? 1 : 0;
-}
-#endif
-
 int
 ISread_n(InputStream stream, char *dst, int count)
 {
