From dbaf87d73d5b134c0eed2893cd43aa548ad0556c Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 21:42:55 +0100
Subject: [PATCH 024/172] Remove unused function

---
 istream.c | 9 ---------
 1 file changed, 9 deletions(-)

--- a/istream.c
+++ b/istream.c
@@ -320,15 +320,6 @@ ISfileno(InputStream stream)
     }
 }
 
-int
-ISeos(InputStream stream)
-{
-    BaseStream base = &stream->base;
-    if (!base->iseos && MUST_BE_UPDATED(base))
-	do_update(base);
-    return base->iseos;
-}
-
 #ifdef USE_SSL
 static Str accept_this_site;
 
