From 421ca64248653652ba2e5ff09191e1a3ee866d0b Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 21:43:28 +0100
Subject: [PATCH 025/172] Remove unused function

---
 main.c | 9 ---------
 1 file changed, 9 deletions(-)

--- a/main.c
+++ b/main.c
@@ -3161,15 +3161,6 @@ DEFUN(followA, GOTO_LINK, "Follow curren
     displayBuffer(Currentbuf, B_NORMAL);
 }
 
-/* follow HREF link in the buffer */
-void
-bufferA(void)
-{
-    on_target = FALSE;
-    followA();
-    on_target = TRUE;
-}
-
 /* view inline image */
 DEFUN(followI, VIEW_IMAGE, "Display image in viewer")
 {
