From d2f5c336234165b44f88e0bf15b1560a9de2180b Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 21:43:43 +0100
Subject: [PATCH 026/172] Remove unused function

---
 main.c | 7 -------
 1 file changed, 7 deletions(-)

--- a/main.c
+++ b/main.c
@@ -3408,13 +3408,6 @@ DEFUN(submitForm, SUBMIT, "Submit form")
     _followForm(TRUE);
 }
 
-/* process form */
-void
-followForm(void)
-{
-    _followForm(FALSE);
-}
-
 static void
 _followForm(int submit)
 {
