From 5a96ed931f680188b8a6c9f212ca1456bd66e14c Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 21:44:03 +0100
Subject: [PATCH 028/172] Remove unused function

---
 mimehead.c | 10 ----------
 1 file changed, 10 deletions(-)

--- a/mimehead.c
+++ b/mimehead.c
@@ -122,16 +122,6 @@ last:
     return;
 }
 
-Str
-decodeU(char **ww)
-{
-    struct growbuf gb;
-
-    growbuf_init(&gb);
-    decodeU_to_growbuf(&gb, ww);
-    return growbuf_to_Str(&gb);
-}
-
 void
 decodeU_to_growbuf(struct growbuf *gb, char **ww)
 {
