From 70391c46049e181b5894c9300f2c647d397b4e60 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 21:44:15 +0100
Subject: [PATCH 029/172] Remove unused function

---
 mimehead.c | 11 -----------
 1 file changed, 11 deletions(-)

--- a/mimehead.c
+++ b/mimehead.c
@@ -173,17 +173,6 @@ decodeQ(char **ww)
     return a;
 }
 
-/* RFC2045 (6.7. Quoted-Printable Content-Transfer-Encoding) */
-Str
-decodeQP(char **ww)
-{
-    struct growbuf gb;
-
-    growbuf_init(&gb);
-    decodeQP_to_growbuf(&gb, ww);
-    return growbuf_to_Str(&gb);
-}
-
 void
 decodeQP_to_growbuf(struct growbuf *gb, char **ww)
 {
