From f8b57fa8e145f73fbfc9e7296cbb9720f8b7b0fe Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 21:57:45 +0100
Subject: [PATCH 031/172] Remove unused function

---
 indep.c | 16 ----------------
 1 file changed, 16 deletions(-)

--- a/indep.c
+++ b/indep.c
@@ -325,22 +325,6 @@ strcasestr(const char *s1, const char *s
 }
 #endif
 
-static int
-strcasematch(char *s1, char *s2)
-{
-    int x;
-    while (*s1) {
-	if (*s2 == '\0')
-	    return 1;
-	x = TOLOWER(*s1) - TOLOWER(*s2);
-	if (x != 0)
-	    break;
-	s1++;
-	s2++;
-    }
-    return (*s2 == '\0');
-}
-
 int
 strmatchlen(const char *s1, const char *s2, int maxlen)
 {
