From baeced479e531ff15db0706af21d51a9accb3dc0 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Wed, 12 Jan 2022 12:30:58 +0100
Subject: [PATCH 040/172] Remove redundant code

---
 display.c | 4 ----
 1 file changed, 4 deletions(-)

--- a/display.c
+++ b/display.c
@@ -498,10 +498,6 @@ displayBuffer(Buffer *buf, int mode)
 	save_current_buf = buf;
     }
 #endif
-    if (mode == B_FORCE_REDRAW &&  (buf->check_url & CHK_URL) ) {
-	chkURLBuffer(buf);
-	displayBuffer(buf, B_NORMAL);
-    }
 }
 
 static void
