From 067e4399b9201a97bc470c2e9be1855464a1bb04 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Wed, 12 Jan 2022 11:21:33 +0100
Subject: [PATCH 041/172] Toggle check_url with chkURL

---
 main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/main.c
+++ b/main.c
@@ -5021,7 +5021,7 @@ DEFUN(defCSet, DEFAULT_CHARSET, "Change
 DEFUN(chkURL, MARK_URL, "Turn URL-like strings into hyperlinks")
 {
     Currentbuf->need_reshape = 1;
-    Currentbuf->check_url = 1;
+    Currentbuf->check_url = !Currentbuf->check_url;
     displayBuffer(Currentbuf, B_FORCE_REDRAW);
 }
 
