From 6499ed95b1cef62de886f7e9a4e71e85c48d5436 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 6 Jan 2022 11:56:31 +0100
Subject: [PATCH 053/172] Dead store

---
 Str.c     |    4 ++--
 inflate.c |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

--- a/inflate.c
+++ b/inflate.c
@@ -61,7 +61,7 @@ main(int argc, char **argv)
 	    }
 	    s.next_in = (Bytef *) dummy_head;
 	    s.avail_in = sizeof(dummy_head);
-	    status = inflate(&s, flush);
+	    inflate(&s, flush);
 	    s.next_in = (Bytef *) inbuf;
 	    s.avail_in = len;
 	    continue;
--- a/Str.c
+++ b/Str.c
@@ -537,7 +537,7 @@ Sprintf(char *fmt, ...)
 		case 'x':
 		case 'X':
 		case 'u':
-		    vi = va_arg(ap, int);
+		    va_arg(ap, int);
 		    len += (p > 0) ? p : 10;
 		    break;
 		case 'f':
@@ -550,7 +550,7 @@ Sprintf(char *fmt, ...)
 		    break;
 		case 'c':
 		    len += 1;
-		    vi = va_arg(ap, int);
+		    va_arg(ap, int);
 		    break;
 		case 's':
 		    vs = va_arg(ap, char *);
