From c97e5457f6d40db1687926d65bff51bd9c330d63 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Sat, 8 Jan 2022 13:26:46 +0100
Subject: [PATCH 081/172] Only use malloc if the default buffer is to small

---
 table.c |   36 ++++++++++++++++++++++++++++--------
 1 file changed, 28 insertions(+), 8 deletions(-)

--- a/table.c
+++ b/table.c
@@ -1256,12 +1256,21 @@ check_table_width(struct table *t, doubl
     struct table_cell *cell = &t->cell;
     int maxcol = t->maxcol >= 0 ? t->maxcol + 1 : 1;
     int maxcell = cell->maxcell >= 0 ? cell->maxcell + 1 : 1;
-    short *orgwidth = GC_malloc(maxcol * sizeof(*orgwidth));
-    short *corwidth = GC_malloc(maxcol * sizeof(*corwidth));
-    short *cwidth = GC_malloc(maxcell * sizeof(*cwidth));
-    double *swidth = GC_malloc(maxcell * sizeof(*swidth));
+    short orgwidth_buf[MAXCOL], *orgwidth = orgwidth_buf;
+    short corwidth_buf[MAXCOL], *corwidth = corwidth_buf;
+    short cwidth_buf[MAXCELL], *cwidth = cwidth_buf;
+    double swidth_buf[MAXCELL], *swidth = swidth_buf;
     double twidth, sxy, *Sxx, stotal;
 
+    if (maxcol > MAXCOL) {
+	orgwidth = GC_malloc(maxcol * sizeof(*orgwidth));
+	corwidth = GC_malloc(maxcol * sizeof(*corwidth));
+    }
+    if (maxcell > MAXCELL) {
+	cwidth = GC_malloc(maxcell * sizeof(*cwidth));
+	swidth = GC_malloc(maxcell * sizeof(*swidth));
+    }
+
     twidth = 0.;
     stotal = 0.;
     for (i = 0; i <= t->maxcol; i++) {
@@ -1605,7 +1614,9 @@ static int
 get_table_width(struct table *t, short *orgwidth, short *cellwidth, int flag)
 {
     int maxcol = t->maxcol >= 0 ? t->maxcol + 1 : 1;
-    short *newwidth = GC_malloc(maxcol * sizeof(*newwidth));
+    short newwidth_buf[MAXCOL], *newwidth = newwidth_buf;
+    if (maxcol > MAXCOL)
+	newwidth = GC_malloc(maxcol * sizeof(*newwidth));
     int i;
     int swidth;
     struct table_cell *cell = &t->cell;
@@ -1616,7 +1627,9 @@ get_table_width(struct table *t, short *
 
     if (flag & CHECK_FIXED) {
 	int maxcell = cell->maxcell >= 0 ? cell->maxcell + 1 : 1;
-	short *ccellwidth = GC_malloc(maxcell * sizeof(*ccellwidth));
+	short ccellwidth_buf[MAXCELL], *ccellwidth = ccellwidth_buf;
+	if (maxcell > MAXCELL)
+	    ccellwidth = GC_malloc(maxcell * sizeof(*ccellwidth));
 	for (i = 0; i <= t->maxcol; i++) {
 	    if (newwidth[i] < t->fixed_width[i])
 		newwidth[i] = t->fixed_width[i];
@@ -3488,8 +3501,15 @@ set_table_matrix0(struct table *t, int m
     int i, j, k, bcol, ecol;
     int width;
     double w0, w1, w, s, b;
-    double *we = GC_malloc(size * sizeof(*we));
-    char *expand = GC_malloc(size * sizeof(*expand));
+
+    double we_buf[MAXCOL], *we = we_buf;
+    char expand_buf[MAXCOL], *expand = expand_buf;
+
+    if (size > MAXCOL) {
+	we = GC_malloc(size * sizeof(*we));
+	expand = GC_malloc(size * sizeof(*expand));
+    }
+
     struct table_cell *cell = &t->cell;
 
     w0 = 0.;
