From 89f28135dd6edeedd561cfda89ac7f3cc968af8e Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Sun, 5 Sep 2021 21:18:16 +0200
Subject: [PATCH 101/172] WIP getch Fix warning (strict-prototype)

---
 terms.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/terms.c
+++ b/terms.c
@@ -2156,6 +2156,7 @@ getch(void)
  * E.g. SELECT_MENU will segfault if you remove it.
  * TODO(rkta): wgetch belongs to libcurses, something fishy is going on here.
  */
+char wgetch(void *); /* TODO(rkta): curses shenanigans... */
 char
 wgetch(void *p)
 {
@@ -2172,6 +2173,7 @@ wgetch(void *p)
     return c;
 }
 
+extern int do_getch(void); /* TODO(rkta): header, found in other file */
 int
 do_getch(void)
 {
