From 9f16d71554c64632dbcb5a946bb7c3d40a3d0569 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 11:55:22 +0100
Subject: [PATCH 135/172] Check return value of freopen

---
 local.c |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

--- a/local.c
+++ b/local.c
@@ -407,19 +407,22 @@ localcgi_post(char *uri, char *qstr, For
 	    set_environ("CONTENT_TYPE",
 			Sprintf("multipart/form-data; boundary=%s",
 				request->boundary)->ptr);
-	    freopen(request->body, "r", stdin);
+	    if (freopen(request->body, "r", stdin) == NULL)
+		return NULL;
 	}
 	else {
 	    set_environ("CONTENT_TYPE", "application/x-www-form-urlencoded");
 	    fwrite(request->body, sizeof(char), request->length, fw);
 	    fclose(fw);
-	    freopen(tmpf, "r", stdin);
+	    if (freopen(tmpf, "r", stdin) == NULL)
+		return NULL;
 	}
     }
     else {
 	set_environ("REQUEST_METHOD", "GET");
 	set_environ("QUERY_STRING", qstr ? qstr : "");
-	freopen(DEV_NULL_PATH, "r", stdin);
+	if (freopen(DEV_NULL_PATH, "r", stdin) == NULL)
+	    return NULL;
     }
 
 #ifdef HAVE_CHDIR		/* ifndef __EMX__ ? */
