From 213cff12a39713c04941aea579362f0f1a1b7191 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 12:22:42 +0100
Subject: [PATCH 137/172] Remove redundant NULL check

---
 url.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/url.c
+++ b/url.c
@@ -576,7 +576,7 @@ openSocket(char *const hostname,
 #ifdef INET6
     /* rfc2732 compliance */
     hname = hostname;
-    if (hname != NULL && hname[0] == '[' && hname[strlen(hname) - 1] == ']') {
+    if (hname[0] == '[' && hname[strlen(hname) - 1] == ']') {
 	hname = allocStr(hostname + 1, -1);
 	hname[strlen(hname) - 1] = '\0';
 	if (strspn(hname, "0123456789abcdefABCDEF:.") != strlen(hname))
