From 59207052154f51306c88e7d0a7bcc0ef6228fac7 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 12:27:10 +0100
Subject: [PATCH 138/172] Remove redundant NULL check

---
 buffer.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/buffer.c
+++ b/buffer.c
@@ -158,7 +158,7 @@ replaceBuffer(Buffer *first, Buffer *del
 	discardBuffer(delbuf);
 	return newbuf;
     }
-    if (delbuf && (buf = prevBuffer(first, delbuf))) {
+    if ((buf = prevBuffer(first, delbuf))) {
 	buf->nextBuffer = newbuf;
 	newbuf->nextBuffer = delbuf->nextBuffer;
 	discardBuffer(delbuf);
