From 10869b3f9b90b58feefb05df0fd8c72446121ad8 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 12:28:54 +0100
Subject: [PATCH 139/172] Remove redundant NULL check

---
 file.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/file.c
+++ b/file.c
@@ -1081,7 +1081,7 @@ extract_auth_val(char **q)
 		}
 	    }
 	}
-	else if (quoted && *qq == '\\')
+	else if (*qq == '\\')
 	    Strcat_char(val, *qq++);
 	Strcat_char(val, *qq++);
     }
