From 3172e472a375518394f84dfa235d37a9bc9dab3d Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Fri, 22 Apr 2022 10:13:10 +0200
Subject: [PATCH 151/172] Fix potential null dereference

---
 news.c | 2 ++
 1 file changed, 2 insertions(+)

--- a/news.c
+++ b/news.c
@@ -116,6 +116,8 @@ name_from_address(char *str, int n)
     int l, space = TRUE;
 
     s = allocStr(str, -1);
+    if (!s)
+	return NULL;
     SKIP_BLANKS(s);
     if (*s == '<' && (p = strchr(s, '>'))) {
 	*p++ = '\0';
