From 603c4577a47c2663de79620988f8a6b3473faa88 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Fri, 22 Apr 2022 10:15:06 +0200
Subject: [PATCH 152/172] Fix potential null dereference

---
 news.c | 3 +++
 1 file changed, 3 insertions(+)

--- a/news.c
+++ b/news.c
@@ -162,6 +162,9 @@ html_quote_s(char *str)
     char *p, *q;
     int space = TRUE;
 
+    if (!str)
+	return NULL;
+
     for (p = str; *p; p++) {
 	if (IS_SPACE(*p)) {
 	    if (space)
