From 6368f9fb4a51802c0bd1d1f0706c2af44adc531f Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 13 Jan 2022 13:28:26 +0100
Subject: [PATCH 155/172] Add more const to anchor

---
 anchor.c |   25 +++++++++++++------------
 proto.h  |    4 ++--
 2 files changed, 15 insertions(+), 14 deletions(-)

--- a/anchor.c
+++ b/anchor.c
@@ -191,7 +191,7 @@ searchURLLabel(Buffer *buf, char *url)
 
 #ifdef USE_NNTP
 static Anchor *
-_put_anchor_news(Buffer *buf, char *p1, const char *p2, int line, int pos)
+_put_anchor_news(Buffer *buf, const char *p1, const char *p2, int line, int pos)
 {
     Str tmp;
 
@@ -210,7 +210,7 @@ _put_anchor_news(Buffer *buf, char *p1,
 #endif				/* USE_NNTP */
 
 static Anchor *
-_put_anchor_all(Buffer *buf, char *p1, const char *p2, int line, int pos)
+_put_anchor_all(Buffer *buf, const char *p1, const char *p2, int line, int pos)
 {
     Str tmp;
 
@@ -294,9 +294,10 @@ reseq_anchor(Buffer *buf)
     reseq_anchor0(buf->formitem, seqmap);
 }
 
-static char *
-reAnchorPos(Buffer *buf, Line *l, char *p1, char *p2,
-	    Anchor *(*anchorproc) (Buffer *, char *, const char *, int, int))
+static const char *
+reAnchorPos(Buffer *buf, Line *l, const char *p1, const char *p2,
+	    Anchor *(*anchorproc) (Buffer *, const char *, const char *,
+				   int, int))
 {
     Anchor *a;
     int spos, epos;
@@ -345,12 +346,12 @@ reAnchorWord(Buffer *buf, Line *l, int s
 
 /* search regexp and register them as anchors */
 /* returns error message if any               */
-static char *
-reAnchorAny(Buffer *buf, char *re,
-	    Anchor *(*anchorproc) (Buffer *, char *, const char *, int, int))
+static const char *
+reAnchorAny(Buffer *buf, const char *re,
+	    Anchor *(*anchorproc) (Buffer *, const char *, const char *, int, int))
 {
     Line *l;
-    char *p = NULL, *p1, *p2;
+    const char *p = NULL, *p1, *p2;
 
     if (re == NULL || *re == '\0') {
 	return NULL;
@@ -380,14 +381,14 @@ reAnchorAny(Buffer *buf, char *re,
     return NULL;
 }
 
-char *
+const char *
 reAnchor(Buffer *buf, char *re)
 {
     return reAnchorAny(buf, re, _put_anchor_all);
 }
 
 #ifdef USE_NNTP
-char *
+const char *
 reAnchorNews(Buffer *buf, char *re)
 {
     return reAnchorAny(buf, re, _put_anchor_news);
@@ -397,7 +398,7 @@ char *
 reAnchorNewsheader(Buffer *buf)
 {
     Line *l;
-    char *p, *p1, *p2;
+    const char *p, *p1, *p2;
     static char *header_mid[] = {
 	"Message-Id:", "References:", "In-Reply-To:", NULL
     };
--- a/proto.h
+++ b/proto.h
@@ -571,9 +571,9 @@ extern Anchor *retrieveCurrentForm(Buffe
 extern Anchor *searchAnchor(AnchorList *al, char *str);
 extern Anchor *searchURLLabel(Buffer *buf, char *url);
 extern void reAnchorWord(Buffer *buf, Line *l, int spos, int epos);
-extern char *reAnchor(Buffer *buf, char *re);
+extern const char *reAnchor(Buffer *buf, char *re);
 #ifdef USE_NNTP
-extern char *reAnchorNews(Buffer *buf, char *re);
+extern const char *reAnchorNews(Buffer *buf, char *re);
 extern char *reAnchorNewsheader(Buffer *buf);
 #endif				/* USE_NNTP */
 extern void addMultirowsForm(Buffer *buf, AnchorList *al);
