From c43eebd9dc7c0857b6f8dcf908e6c31740953418 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 13 Jan 2022 13:29:40 +0100
Subject: [PATCH 156/172] Add more const to linein

---
 linein.c |    3 ++-
 proto.h  |    5 +++--
 2 files changed, 5 insertions(+), 3 deletions(-)

--- a/linein.c
+++ b/linein.c
@@ -80,7 +80,8 @@ static void ins_char(Str str);
 #endif
 
 char *
-inputLineHistSearch(char *prompt, char *def_str, int flag, Hist *hist,
+inputLineHistSearch(const char *prompt, const char *def_str, int flag,
+		    Hist *hist,
 		    int (*incrfunc) (int ch, Str str, Lineprop *prop))
 {
     int opos, x, y, lpos, rpos, epos;
--- a/proto.h
+++ b/proto.h
@@ -375,8 +375,9 @@ extern void escmap(void);
 extern void escbmap(void);
 extern void escdmap(char c);
 extern void multimap(void);
-extern char *inputLineHistSearch(char *prompt, char *def_str, int flag,
-				 Hist *hist, int (*incfunc) (int ch, Str buf,
+extern char *inputLineHistSearch(const char *prompt, const char *def_str,
+				 int flag, Hist *hist,
+				 int (*incfunc) (int ch, Str buf,
 							     Lineprop *prop));
 extern Str unescape_spaces(Str s);
 #ifdef USE_HISTORY
