From 9d5c211fe32740d759914a1aa93da13d608f749b Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 13 Jan 2022 13:30:54 +0100
Subject: [PATCH 157/172] Add more const to search

---
 proto.h  |    6 +++---
 search.c |   16 ++++++++--------
 2 files changed, 11 insertions(+), 11 deletions(-)

--- a/proto.h
+++ b/proto.h
@@ -361,12 +361,12 @@ extern Str correct_irrtag(int status);
 extern void init_migemo(void);
 #endif
 #ifdef USE_M17N
-extern char *conv_search_string(char *str, wc_ces f_ces);
+extern const char *conv_search_string(const char *str, wc_ces f_ces);
 #else
 #define conv_search_string(str, f_ces)	str
 #endif
-extern int forwardSearch(Buffer *buf, char *str);
-extern int backwardSearch(Buffer *buf, char *str);
+extern int forwardSearch(Buffer *buf, const char *str);
+extern int backwardSearch(Buffer *buf, const char *str);
 extern void pcmap(void);
 extern void escmap(void);
 extern void escbmap(void);
--- a/search.c
+++ b/search.c
@@ -51,8 +51,8 @@ open_migemo(char *migemo_command)
     return 0;
 }
 
-static char *
-migemostr(char *str)
+static const char *
+migemostr(const char *str)
 {
     Str tmp = NULL;
     if (migemor == NULL || migemow == NULL)
@@ -83,8 +83,8 @@ migemostr(char *str)
 
 #ifdef USE_M17N
 /* normalize search string */
-char *
-conv_search_string(char *str, wc_ces f_ces)
+const char *
+conv_search_string(const char *str, wc_ces f_ces)
 {
     if (SearchConv && !WcOption.pre_conv &&
 	Currentbuf->document_charset != f_ces)
@@ -94,9 +94,9 @@ conv_search_string(char *str, wc_ces f_c
 #endif
 
 int
-forwardSearch(Buffer *buf, char *str)
+forwardSearch(Buffer *buf, const char *str)
 {
-    char *p, *first, *last;
+    const char *p, *first, *last;
     Line *l, *begin;
     int wrapped = FALSE;
     int pos;
@@ -189,9 +189,9 @@ forwardSearch(Buffer *buf, char *str)
 }
 
 int
-backwardSearch(Buffer *buf, char *str)
+backwardSearch(Buffer *buf, const char *str)
 {
-    char *p, *q, *found, *found_last, *first, *last;
+    const char *p, *q, *found, *found_last, *first, *last;
     Line *l, *begin;
     int wrapped = FALSE;
     int pos;
