From da95128c88ec5867f8bda01ecb64b98475adc07a Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 13 Jan 2022 13:32:01 +0100
Subject: [PATCH 158/172] Add more const to menu

---
 menu.c |   20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

--- a/menu.c
+++ b/menu.c
@@ -924,15 +924,15 @@ mSusp(char c)
     return (MENU_NOTHING);
 }
 
-static char *SearchString = NULL;
+static const char *SearchString = NULL;
 
-int (*menuSearchRoutine) (Menu *, char *, int);
+int (*menuSearchRoutine) (Menu *, const char *, int);
 
 static int
-menuForwardSearch(Menu *menu, char *str, int from)
+menuForwardSearch(Menu *menu, const char *str, int from)
 {
     int i;
-    char *p;
+    const char *p;
     if ((p = regexCompile(str, IgnoreCase)) != NULL) {
 	message(p, 0, 0);
 	return -1;
@@ -949,7 +949,7 @@ menuForwardSearch(Menu *menu, char *str,
 static int
 menu_search_forward(Menu *menu, int from)
 {
-    char *str;
+    const char *str;
     int found;
     str = inputStrHist("Forward: ", NULL, TextHist);
     if (str != NULL && *str == '\0')
@@ -979,10 +979,10 @@ mSrchF(char c)
 }
 
 static int
-menuBackwardSearch(Menu *menu, char *str, int from)
+menuBackwardSearch(Menu *menu, const char *str, int from)
 {
     int i;
-    char *p;
+    const char *p;
     if ((p = regexCompile(str, IgnoreCase)) != NULL) {
 	message(p, 0, 0);
 	return -1;
@@ -999,7 +999,7 @@ menuBackwardSearch(Menu *menu, char *str
 static int
 menu_search_backward(Menu *menu, int from)
 {
-    char *str;
+    const char *str;
     int found;
     str = inputStrHist("Backward: ", NULL, TextHist);
     if (str != NULL && *str == '\0')
@@ -1032,9 +1032,9 @@ static int
 menu_search_next_previous(Menu *menu, int from, int reverse)
 {
     int found;
-    static int (*routine[2]) (Menu *, char *, int) = {
+    static int (*routine[2]) (Menu *, const char *, int) = {
     menuForwardSearch, menuBackwardSearch};
-    char *str;
+    const char *str;
 
     if (menuSearchRoutine == NULL) {
 	disp_message("No previous regular expression", TRUE);
