From aa34b4924a819018f790ff9a02848a9db68d7096 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 13 Jan 2022 13:32:21 +0100
Subject: [PATCH 159/172] Add more const to rc

---
 rc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/rc.c
+++ b/rc.c
@@ -1875,7 +1875,7 @@ querySiteconf(ParsedURL *query_pu, int f
 {
     struct siteconf_rec *ent;
     Str u;
-    char *firstp, *lastp;
+    const char *firstp, *lastp;
 
     if (field < 0 || field >= SCONF_N_FIELD)
 	return NULL;
