From a40ad6cee45a54997093ec3cf6dbd55af3066770 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 13 Jan 2022 13:33:43 +0100
Subject: [PATCH 161/172] Add more const to main

---
 main.c |   26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

--- a/main.c
+++ b/main.c
@@ -80,10 +80,10 @@ static MySignalHandler SigPipe(SIGNAL_AR
 #endif
 
 #ifdef USE_MARK
-static char *MarkString = NULL;
+static const char *MarkString = NULL;
 #endif
-static char *SearchString = NULL;
-int (*searchRoutine) (Buffer *, char *);
+static const char *SearchString = NULL;
+int (*searchRoutine) (Buffer *, const char *);
 
 #ifndef __MINGW32_VERSION
 JMP_BUF IntReturn;
@@ -1784,7 +1784,7 @@ clear_mark(Line *l)
 
 /* search by regular expression */
 static int
-srchcore(char *volatile str, int (*func) (Buffer *, char *))
+srchcore(const char *str, int (*func) (Buffer *, const char *))
 {
     MySignalHandler(*prevtrap) (SIGNAL_ARG);
     volatile int i, result = SR_NOTFOUND;
@@ -1810,7 +1810,7 @@ srchcore(char *volatile str, int (*func)
 }
 
 static void
-disp_srchresult(int result, char *prompt, char *str)
+disp_srchresult(int result, char *prompt, const char *str)
 {
     if (str == NULL)
 	str = "";
@@ -1895,7 +1895,7 @@ dispincsrch(int ch, Str buf, Lineprop *p
 }
 
 static void
-isrch(int (*func) (Buffer *, char *), char *prompt)
+isrch(int (*func) (Buffer *, const char *), char *prompt)
 {
     char *str;
     Buffer sbuf;
@@ -1911,9 +1911,9 @@ isrch(int (*func) (Buffer *, char *), ch
 }
 
 static void
-srch(int (*func) (Buffer *, char *), char *prompt)
+srch(int (*func) (Buffer *, const char *), char *prompt)
 {
-    char *str;
+    const char *str;
     int result;
     int disp = FALSE;
     int pos;
@@ -1972,7 +1972,7 @@ srch_nxtprv(int reverse)
 {
     int result;
     /* *INDENT-OFF* */
-    static int (*routine[2]) (Buffer *, char *) = {
+    static int (*routine[2]) (Buffer *, const char *) = {
 	forwardSearch, backwardSearch
     };
     /* *INDENT-ON* */
@@ -2404,9 +2404,9 @@ DEFUN(movR1, MOVE_RIGHT1, "Cursor right.
 #define prevChar(s, l)	do { (s)--; } while ((s) > 0 && (l)->propBuf[s] & PC_WCHAR2)
 
 static wc_uint32
-getChar(char *p)
+getChar(const char *p)
 {
-    return wc_any_to_ucs(wtf_parse1((wc_uchar **)&p));
+    return wc_any_to_ucs(wtf_parse1((const wc_uchar **)&p));
 }
 
 static int
@@ -2931,8 +2931,8 @@ DEFUN(prevMk, PREV_MARK, "Go to the prev
 DEFUN(reMark, REG_MARK, "Mark all occurences of a pattern")
 {
     Line *l;
-    char *str;
-    char *p, *p1, *p2;
+    const char *str;
+    const char *p, *p1, *p2;
 
     if (!use_mark)
 	return;
