From 5e5bb74929f579dc60fe27f932d6db2e456f6164 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 13 Jan 2022 13:34:21 +0100
Subject: [PATCH 162/172] Add more const to display

---
 display.c |    6 +++---
 proto.h   |    8 ++++----
 2 files changed, 7 insertions(+), 7 deletions(-)

--- a/display.c
+++ b/display.c
@@ -1236,7 +1236,7 @@ message_list_panel(void)
 }
 
 void
-message(char *s, int return_x, int return_y)
+message(const char *s, int return_x, int return_y)
 {
     if (!fmInitialized)
 	return;
@@ -1254,7 +1254,7 @@ disp_err_message(char *s, int redraw_cur
 }
 
 void
-disp_message_nsec(char *s, int redraw_current, int sec, int purge, int mouse)
+disp_message_nsec(const char *s, int redraw_current, int sec, int purge, int mouse)
 {
     if (QuietMessage)
 	return;
@@ -1282,7 +1282,7 @@ disp_message_nsec(char *s, int redraw_cu
 }
 
 void
-disp_message(char *s, int redraw_current)
+disp_message(const char *s, int redraw_current)
 {
     disp_message_nsec(s, redraw_current, 10, FALSE, TRUE);
 }
--- a/proto.h
+++ b/proto.h
@@ -318,11 +318,11 @@ extern void addMChar(char *c, Lineprop m
 #endif
 extern void record_err_message(char *s);
 extern Buffer *message_list_panel(void);
-extern void message(char *s, int return_x, int return_y);
+extern void message(const char *s, int return_x, int return_y);
 extern void disp_err_message(char *s, int redraw_current);
-extern void disp_message_nsec(char *s, int redraw_current, int sec, int purge,
-			      int mouse);
-extern void disp_message(char *s, int redraw_current);
+extern void disp_message_nsec(const char *s, int redraw_current, int sec,
+			      int purge, int mouse);
+extern void disp_message(const char *s, int redraw_current);
 #ifdef USE_MOUSE
 extern void disp_message_nomouse(char *s, int redraw_current);
 #else
