From 277d6935d04898b99c87ff78aa674f1a5a61a996 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 23 Dec 2021 22:01:05 +0100
Subject: [PATCH 163/172] Add more const const correctness

---
 anchor.c |   17 ++++++++---------
 1 file changed, 8 insertions(+), 9 deletions(-)

--- a/anchor.c
+++ b/anchor.c
@@ -191,7 +191,7 @@ searchURLLabel(Buffer *buf, char *url)
 
 #ifdef USE_NNTP
 static Anchor *
-_put_anchor_news(Buffer *buf, const char *p1, const char *p2, int line, int pos)
+_put_anchor_news(Buffer *buf, char *p1, const char *p2, int line, int pos)
 {
     Str tmp;
 
@@ -210,7 +210,7 @@ _put_anchor_news(Buffer *buf, const char
 #endif				/* USE_NNTP */
 
 static Anchor *
-_put_anchor_all(Buffer *buf, const char *p1, const char *p2, int line, int pos)
+_put_anchor_all(Buffer *buf, char *p1, const char *p2, int line, int pos)
 {
     Str tmp;
 
@@ -294,10 +294,9 @@ reseq_anchor(Buffer *buf)
     reseq_anchor0(buf->formitem, seqmap);
 }
 
-static const char *
-reAnchorPos(Buffer *buf, Line *l, const char *p1, const char *p2,
-	    Anchor *(*anchorproc) (Buffer *, const char *, const char *,
-				   int, int))
+static char *
+reAnchorPos(Buffer *buf, Line *l, char *p1, char *p2,
+	    Anchor *(*anchorproc) (Buffer *, char *, const char *, int, int))
 {
     Anchor *a;
     int spos, epos;
@@ -346,9 +345,9 @@ reAnchorWord(Buffer *buf, Line *l, int s
 
 /* search regexp and register them as anchors */
 /* returns error message if any               */
-static const char *
-reAnchorAny(Buffer *buf, const char *re,
-	    Anchor *(*anchorproc) (Buffer *, const char *, const char *, int, int))
+static char *
+reAnchorAny(Buffer *buf, char *re,
+	    Anchor *(*anchorproc) (Buffer *, char *, const char *, int, int))
 {
     Line *l;
     const char *p = NULL, *p1, *p2;
