From 636fc659a747a5bcc84979ff7084fd56245c288e Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Sat, 26 Nov 2022 10:34:37 +0100
Subject: [PATCH 171/172] Add more const correctness

---
 etc.c   |    4 ++--
 proto.h |    6 +++---
 url.c   |    4 ++--
 3 files changed, 7 insertions(+), 7 deletions(-)

--- a/etc.c
+++ b/etc.c
@@ -1686,10 +1686,10 @@ file_to_url(char *file)
 
 #ifdef USE_M17N
 char *
-url_unquote_conv(char *url, wc_ces charset)
+url_unquote_conv(const char *url, wc_ces charset)
 #else
 char *
-url_unquote_conv0(char *url)
+url_unquote_conv0(const char *url)
 #endif
 {
 #ifdef USE_M17N
--- a/proto.h
+++ b/proto.h
@@ -505,7 +505,7 @@ extern int openSocket(char *hostname, ch
 extern void parseURL(char *url, ParsedURL *p_url, ParsedURL *current);
 extern void copyParsedURL(ParsedURL *p, const ParsedURL *q);
 extern void parseURL2(char *url, ParsedURL *pu, ParsedURL *current);
-extern Str parsedURL2Str(ParsedURL *pu);
+extern Str parsedURL2Str(const ParsedURL *pu);
 extern Str parsedURL2RefererStr(ParsedURL *pu);
 extern int getURLScheme(char **url);
 extern void init_stream(URLFile *uf, int scheme, InputStream stream);
@@ -634,9 +634,9 @@ extern Str myEditor(char *cmd, char *fil
 extern int is_localhost(const char *host);
 extern char *file_to_url(char *file);
 #ifdef USE_M17N
-extern char *url_unquote_conv(char *url, wc_ces charset);
+extern char *url_unquote_conv(const char *url, wc_ces charset);
 #else
-extern char *url_unquote_conv0(char *url);
+extern char *url_unquote_conv0(const char *url);
 #define url_unquote_conv(url, charset) url_unquote_conv0(url)
 #endif
 extern char *expandName(char *name);
--- a/url.c
+++ b/url.c
@@ -1247,7 +1247,7 @@ parseURL2(char *url, ParsedURL *pu, Pars
 }
 
 static Str
-_parsedURL2Str(ParsedURL *pu, int pass, int user, int label)
+_parsedURL2Str(const ParsedURL *pu, int pass, int user, int label)
 {
     Str tmp;
     static char *scheme_str[] = {
@@ -1341,7 +1341,7 @@ _parsedURL2Str(ParsedURL *pu, int pass,
 }
 
 Str
-parsedURL2Str(ParsedURL *pu)
+parsedURL2Str(const ParsedURL *pu)
 {
     return _parsedURL2Str(pu, FALSE, TRUE, TRUE);
 }
