From: Rene Kita <mail@rkta.de>
Date: Sun Oct 29 16:11:29 CET 2023
Subject: add gophers support
---
 file.c |    4 ++--
 html.h |    7 ++++++-
 main.c |    2 +-
 url.c  |   40 +++++++++++++++++++++++++++++++++-------
 4 files changed, 42 insertions(+), 11 deletions(-)

--- a/file.c
+++ b/file.c
@@ -1951,7 +1951,7 @@ loadGeneralFile(char *path, ParsedURL *v
     }
 #endif				/* USE_NNTP */
 #ifdef USE_GOPHER
-    else if (pu.scheme == SCM_GOPHER) {
+    else if (pu.scheme == SCM_GOPHER || pu.scheme == SCM_GOPHERS) {
 	p = pu.file;
 	while(*p == '/')
 	    ++p;
@@ -2125,7 +2125,7 @@ loadGeneralFile(char *path, ParsedURL *v
 		return NULL;
 	    file = guess_filename(pu.file);
 #ifdef USE_GOPHER
-	    if (f.scheme == SCM_GOPHER)
+	    if (f.scheme == SCM_GOPHER || f.scheme == SCM_GOPHERS)
 		file = Sprintf("%s.html", file)->ptr;
 #endif
 #ifdef USE_NNTP
--- a/url.c
+++ b/url.c
@@ -72,6 +72,7 @@ static int
     119,			/* news group */
     0,				/* data - not defined */
     0,				/* mailto - not defined */
+    70,				/* gophers */
 #ifdef USE_SSL
     443,			/* https */
 #endif				/* USE_SSL */
@@ -80,6 +81,7 @@ static int
 struct cmdtable schemetable[] = {
     {"http", SCM_HTTP},
     {"gopher", SCM_GOPHER},
+    {"gophers", SCM_GOPHERS},
     {"ftp", SCM_FTP},
     {"local", SCM_LOCAL},
     {"file", SCM_LOCAL},
@@ -229,6 +231,7 @@ DefaultFile(int scheme)
 	return allocStr(HTTP_DEFAULT_FILE, -1);
 #ifdef USE_GOPHER
     case SCM_GOPHER:
+    case SCM_GOPHERS:
 	return allocStr("1", -1);
 #endif				/* USE_GOPHER */
     case SCM_LOCAL:
@@ -962,7 +965,8 @@ parseURL(char *url, ParsedURL *p_url, Pa
 
     q = p;
 #ifdef USE_GOPHER
-    if (p_url->scheme == SCM_GOPHER) {
+    if (p_url->scheme == SCM_GOPHER
+	|| p_url->scheme == SCM_GOPHERS) {
 	if (*q == '/')
 	    q++;
 	if (*q && q[0] != '/' && q[1] != '/' && q[2] == '/')
@@ -976,7 +980,8 @@ parseURL(char *url, ParsedURL *p_url, Pa
 	goto do_query;
     }
 #ifdef USE_GOPHER
-    if (p_url->scheme == SCM_GOPHER && *p == 'R') {
+    if ((p_url->scheme == SCM_GOPHER
+	 || p_url->scheme == SCM_GOPHERS)&& *p == 'R') {
 	if (!*++p) {
 	    p_url->file = "";
 	    goto do_query;
@@ -1145,7 +1150,8 @@ parseURL2(char *url, ParsedURL *pu, Pars
 #endif
 		if (
 #ifdef USE_GOPHER
-		       pu->scheme != SCM_GOPHER &&
+		       (pu->scheme != SCM_GOPHER
+			&& pu->scheme != SCM_GOPHERS) &&
 #endif				/* USE_GOPHER */
 		       pu->file[0] != '/'
 #ifdef SUPPORT_DOS_DRIVE_PREFIX
@@ -1168,7 +1174,8 @@ parseURL2(char *url, ParsedURL *pu, Pars
 		}
 	    }
 #ifdef USE_GOPHER
-	    else if (pu->scheme == SCM_GOPHER && pu->file[0] == '/') {
+	    else if ((pu->scheme == SCM_GOPHER
+		      || pu->scheme == SCM_GOPHERS) && pu->file[0] == '/') {
 		p = pu->file;
 		pu->file = allocStr(p + 1, -1);
 	    }
@@ -1224,7 +1231,8 @@ parseURL2(char *url, ParsedURL *pu, Pars
 	}
 	else if (
 #ifdef USE_GOPHER
-		    pu->scheme != SCM_GOPHER &&
+		    (pu->scheme != SCM_GOPHER
+		     && pu->scheme != SCM_GOPHERS) &&
 #endif				/* USE_GOPHER */
 		    pu->file[0] == '/') {
 	    /*
@@ -1254,9 +1262,11 @@ static Str
 _parsedURL2Str(ParsedURL *pu, int pass, int user, int label)
 {
     Str tmp;
+    /* See SCM_* defines in html.h for correct order of entries. */
     static char *scheme_str[] = {
 	"http", "gopher", "ftp", "ftp", "file", "file", "exec", "nntp", "nntp",
 	"news", "news", "data", "mailto",
+	"gophers",
 #ifdef USE_SSL
 	"https",
 #endif				/* USE_SSL */
@@ -1897,6 +1907,7 @@ openURL(char *url, ParsedURL *pu, Parsed
 	return uf;
 #ifdef USE_GOPHER
     case SCM_GOPHER:
+    case SCM_GOPHERS:
 	p = pu->file;
 	n = 0;
 	while(*p == '/') {
@@ -1955,11 +1966,25 @@ openURL(char *url, ParsedURL *pu, Parsed
 	    tmp = Strnew_charp(file_unquote(pu->file));
 	    Strcat_charp(tmp, "\r\n");
 	}
-	write(sock, tmp->ptr, tmp->length);
+#ifdef USE_SSL
+	if (pu->scheme == SCM_GOPHERS) {
+	    if (!(sslh = openSSLHandle(sock, pu->host,
+				       &uf.ssl_certificate))) {
+		*status = HTST_MISSING;
+		return uf;
+	    }
+	    uf.stream = newSSLStream(sslh, sock);
+	    SSL_write(sslh, tmp->ptr, tmp->length);
+	}
+	else
+#endif
+	{
+	    write(sock, tmp->ptr, tmp->length);
+	    uf.stream = newInputStream(sock);
+	}
 	if(type != '\0') {
 	  pu->file = gophertmp->ptr;
 	}
-	uf.stream = newInputStream(sock);
 	return uf;
 #endif				/* USE_GOPHER */
 #ifdef USE_NNTP
@@ -2414,6 +2439,7 @@ schemeToProxy(int scheme)
 	break;
 #ifdef USE_GOPHER
     case SCM_GOPHER:
+    case SCM_GOPHERS:
 	pu = &GOPHER_proxy_parsed;
 	break;
 #endif
--- a/html.h
+++ b/html.h
@@ -401,6 +401,10 @@ struct environment {
 
 #define INDENT_INCR IndentIncr
 
+/*
+ * Those SCM_ define the indeces of DefaultPort in url.c and scheme_str[] in
+ * _parsedURL2Str in url.c
+ */
 #define SCM_UNKNOWN	255
 #define SCM_MISSING	254
 #define SCM_HTTP	0
@@ -416,8 +420,9 @@ struct environment {
 #define SCM_NEWS_GROUP	10
 #define SCM_DATA	11
 #define SCM_MAILTO      12
+#define SCM_GOPHERS	13
 #ifdef USE_SSL
-#define SCM_HTTPS       13
+#define SCM_HTTPS       14
 #endif				/* USE_SSL */
 
 #endif				/* _HTML_H */
--- a/main.c
+++ b/main.c
@@ -5170,7 +5170,7 @@ chkURLBuffer(Buffer *buf)
 	"https?://[a-zA-Z0-9][a-zA-Z0-9:%\\-\\./?=~_\\&+@#,\\$;]*[a-zA-Z0-9_/=\\-]",
 	"file:/[a-zA-Z0-9:%\\-\\./=_\\+@#,\\$;]*",
 #ifdef USE_GOPHER
-	"gopher://[a-zA-Z0-9][a-zA-Z0-9:%\\-\\./_~]*",
+	"gophers?://[a-zA-Z0-9][a-zA-Z0-9:%\\-\\./_~]*",
 #endif				/* USE_GOPHER */
 	"ftp://[a-zA-Z0-9][a-zA-Z0-9:%\\-\\./=_+@#,\\$]*[a-zA-Z0-9_/]",
 #ifdef USE_NNTP
