From abd25352be4e7484ac07b842ed5976092a6a56de Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 26 May 2022 12:35:11 +0200
Subject: [PATCH 083/172] Add an include guard for proto.h

---
 proto.h |    3 +++
 1 file changed, 3 insertions(+)

--- a/proto.h
+++ b/proto.h
@@ -5,6 +5,8 @@
  *
  *   Created: Wed Feb 10 12:47:03 1999
  */
+#ifndef W3M_PROTO_H__
+#define W3M_PROTO_H__
 extern int main(int argc, char **argv);
 extern void nulcmd(void);
 extern void pushEvent(int cmd, void *data);
@@ -828,3 +830,4 @@ long lrand48(void);
 extern Str base64_encode(const char *src, size_t len);
 
 extern void userMessage(void);
+#endif
