From: Rene Kita <mail@rkta.de>
Date: Tue Sep 17 17:37:57 CEST 2024

---
 Str.c |    8 ++++++++
 Str.h |    1 +
 2 files changed, 9 insertions(+)

--- a/Str.h
+++ b/Str.h
@@ -47,6 +47,7 @@ Str Strcat_m_charp(Str, ...);
 Str Strsubstr(Str, int, int);
 Str Strinsert_char(Str, int, char);
 Str Strinsert_charp(Str, int, const char *);
+Str Strinsert_charp_n(Str s, int pos, const char *p, int n);
 Str Strdelete(Str, int, int);
 Str Strtruncate(Str, int);
 Str Strlower(Str);
--- a/Str.c
+++ b/Str.c
@@ -293,6 +293,14 @@ Strinsert_char(Str s, int pos, char c)
 }
 
 Str
+Strinsert_charp_n(Str s, int pos, const char *p, int n)
+{
+    while (n--)
+	Strinsert_char(s, pos++, *(p++));
+    return s;
+}
+
+Str
 Strinsert_charp(Str s, int pos, const char *p)
 {
     while (*p)
