Author: Rene Kita <mail@rkta.de>
Date: Sun Nov 19 09:16:52 CET 2023
Subject: Check for underflow in checkType
---
 etc.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/etc.c
+++ b/etc.c
@@ -318,6 +318,7 @@ checkType(Str s, Lineprop **oprop, Linec
 	    Strcat_charp_n(s, sp, (int)(str - sp));
 	}
     }
+
     if (!do_copy) {
 	for (; str < endp && IS_ASCII(*str); str++) {
 	    *(prop++) = PE_NORMAL | (IS_CNTRL(*str) ? PC_CTRL : PC_ASCII);
@@ -334,7 +335,7 @@ checkType(Str s, Lineprop **oprop, Linec
     while (str < endp) {
 	if (prop - prop_buffer >= prop_size)
 	    break;
-	if (bs != NULL) {
+	if (bs) {
 #ifdef USE_M17N
 	    if (str == bs - 2 && !strncmp(str, "__\b\b", 4)) {
 		str += 4;
@@ -343,7 +344,6 @@ checkType(Str s, Lineprop **oprop, Linec
 		    bs = memchr(str, '\b', endp - str);
 		continue;
 	    }
-	    else
 #endif
 	    if (str == bs - 1 && *str == '_') {
 		str += 2;
@@ -458,6 +458,7 @@ checkType(Str s, Lineprop **oprop, Linec
 		bs = memchr(str, '\b', endp - str);
 #endif
 	}
+
 #ifdef USE_ANSI_COLOR
 	if (es != NULL) {
 	    if (str == es) {
