From: Rene Kita <mail@rkta.de>
Date: Sat Oct 28 12:06:21 CEST 2023
Subject: Drop support for old ss family
---
 acinclude.m4 |   17 +++--------------
 config.h.in  |    1 -
 configure    |   31 ++-----------------------------
 ftp.c        |    4 ----
 4 files changed, 5 insertions(+), 48 deletions(-)

--- a/acinclude.m4
+++ b/acinclude.m4
@@ -877,20 +877,9 @@ if test x"$enable_ipv6" = xyes; then
 	int i = ss.ss_family;
       ],
       [AC_MSG_RESULT(yes)],
-      [AC_TRY_COMPILE([
-#include <sys/types.h>
-#include <sys/socket.h>
-	],
-	[
-	struct sockaddr_storage ss;
-	int i = ss.__ss_family;
-	],
-	[AC_MSG_RESULT(no, but __ss_family exists)
-	 AC_DEFINE(HAVE_OLD_SS_FAMILY)],
-	[AC_MSG_RESULT(no)
-	 AC_MSG_WARN(IPv6 support is disabled)
-	 enable_ipv6="no"])
-      ])
+      [AC_MSG_RESULT(no)
+      AC_MSG_WARN(IPv6 support is disabled)
+      enable_ipv6="no"])
  fi
  if test x"$enable_ipv6" = xyes; then
     AC_DEFINE(INET6)
--- a/config.h.in
+++ b/config.h.in
@@ -86,7 +86,6 @@
 #undef CLEAR_BUF
 #undef INET6
 #undef HAVE_SOCKLEN_T
-#undef HAVE_OLD_SS_FAMILY
 #undef USE_EGD
 #define ENABLE_REMOVE_TRAILINGSPACES
 #undef MENU_THIN_FRAME
--- a/ftp.c
+++ b/ftp.c
@@ -230,11 +230,7 @@ ftp_pasv(FTP ftp)
     if (getpeername(fileno(ftp->wf),
 		    (struct sockaddr *)&sockaddr, &sockaddrlen) < 0)
 	return -1;
-#ifdef HAVE_OLD_SS_FAMILY
-    family = sockaddr.__ss_family;
-#else
     family = sockaddr.ss_family;
-#endif
 #else
     family = AF_INET;
 #endif
--- a/configure
+++ b/configure
@@ -9256,38 +9256,11 @@ then :
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 printf "%s\n" "yes" >&6; }
 else $as_nop
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-#include <sys/types.h>
-#include <sys/socket.h>
-
-int
-main (void)
-{
-
-	struct sockaddr_storage ss;
-	int i = ss.__ss_family;
-
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"
-then :
   { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
 printf "%s\n" "no" >&6; }
-	 printf "%s\n" "#define HAVE_OLD_SS_FAMILY 1" >>confdefs.h
-
-else $as_nop
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
-printf "%s\n" "no" >&6; }
-	 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: WARNING: IPv6 support is disabled" >&5
+      { printf "%s\n" "$as_me:${as_lineno-$LINENO}: WARNING: IPv6 support is disabled" >&5
 printf "%s\n" "$as_me: WARNING: IPv6 support is disabled" >&2;}
-	 enable_ipv6="no"
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.beam conftest.$ac_ext
-
+      enable_ipv6="no"
 fi
 rm -f core conftest.err conftest.$ac_objext conftest.beam conftest.$ac_ext
  fi
