--- a/Makefile.in
+++ b/Makefile.in
@@ -42,6 +42,7 @@ DEFAULTS += -Wcast-qual
 DEFAULTS += -Wduplicated-branches
 DEFAULTS += -Wduplicated-cond
 DEFAULTS += -Werror=vla
+DEFAULTS += -Wformat=2
 DEFAULTS += -Wlogical-op
 DEFAULTS += -Wmissing-declarations
 DEFAULTS += -Wnested-externs
--- a/libwc/Makefile.in
+++ b/libwc/Makefile.in
@@ -13,6 +13,7 @@ DEFAULTS += -Wcast-qual
 DEFAULTS += -Wduplicated-branches
 DEFAULTS += -Wduplicated-cond
 DEFAULTS += -Werror=vla
+DEFAULTS += -Wformat=2
 DEFAULTS += -Wlogical-op
 DEFAULTS += -Wmissing-declarations
 DEFAULTS += -Wnested-externs
--- a/w3mimg/Makefile.in
+++ b/w3mimg/Makefile.in
@@ -13,6 +13,7 @@ DEFAULTS += -Wcast-qual
 DEFAULTS += -Wduplicated-branches
 DEFAULTS += -Wduplicated-cond
 DEFAULTS += -Werror=vla
+DEFAULTS += -Wformat=2
 DEFAULTS += -Wlogical-op
 DEFAULTS += -Wmissing-declarations
 DEFAULTS += -Wold-style-declaration
--- a/w3mimg/fb/Makefile.in
+++ b/w3mimg/fb/Makefile.in
@@ -14,6 +14,7 @@ DEFAULTS += -Wcast-qual
 DEFAULTS += -Wduplicated-branches
 DEFAULTS += -Wduplicated-cond
 DEFAULTS += -Werror=vla
+DEFAULTS += -Wformat=2
 DEFAULTS += -Wlogical-op
 DEFAULTS += -Wmissing-declarations
 DEFAULTS += -Wold-style-declaration
--- a/w3mimg/win/Makefile.in
+++ b/w3mimg/win/Makefile.in
@@ -14,6 +14,7 @@ DEFAULTS += -Wcast-qual
 DEFAULTS += -Wduplicated-branches
 DEFAULTS += -Wduplicated-cond
 DEFAULTS += -Werror=vla
+DEFAULTS += -Wformat=2
 DEFAULTS += -Wlogical-op
 DEFAULTS += -Wmissing-declarations
 DEFAULTS += -Wold-style-declaration
--- a/w3mimg/x11/Makefile.in
+++ b/w3mimg/x11/Makefile.in
@@ -14,6 +14,7 @@ DEFAULTS += -Wcast-qual
 DEFAULTS += -Wduplicated-branches
 DEFAULTS += -Wduplicated-cond
 DEFAULTS += -Werror=vla
+DEFAULTS += -Wformat=2
 DEFAULTS += -Wlogical-op
 DEFAULTS += -Wmissing-declarations
 DEFAULTS += -Wold-style-declaration
