---
 etc.c   |    5 +++--
 file.c  |    4 ++--
 indep.c |    2 +-
 3 files changed, 6 insertions(+), 5 deletions(-)

--- a/file.c
+++ b/file.c
@@ -8033,7 +8033,7 @@ static Buffer *
 loadcmdout(char *cmd,
 	   Buffer *(*loadproc) (URLFile *, Buffer *), Buffer *defaultbuf)
 {
-    FILE *f, *popen(const char *, const char *);
+    FILE *f;
     Buffer *buf;
     URLFile uf;
 
@@ -8071,7 +8071,7 @@ getshell(char *cmd)
 Buffer *
 getpipe(char *cmd)
 {
-    FILE *f, *popen(const char *, const char *);
+    FILE *f;
     Buffer *buf;
 
     if (cmd == NULL || *cmd == '\0')
--- a/etc.c
+++ b/etc.c
@@ -35,6 +35,8 @@ struct auth_pass {
 
 struct auth_pass *passwords = NULL;
 
+extern Hash_si tagtable;
+
 int
 columnSkip(Buffer *buf, int offset)
 {
@@ -122,7 +124,6 @@ currentLineSkip(Buffer *buf, Line *line,
 int
 gethtmlcmd(char *s)
 {
-    extern Hash_si tagtable;
     char cmdstr[MAX_CMD_LEN];
     char *p = cmdstr;
     char *save = s;
@@ -1600,7 +1601,7 @@ char *
 expandName(char *name)
 {
     char *p;
-    struct passwd *passent, *getpwnam(const char *);
+    struct passwd *passent;
     Str extpath = NULL;
 
     if (name == NULL)
--- a/indep.c
+++ b/indep.c
@@ -183,7 +183,7 @@ char *
 expandPath(char *name)
 {
     char *p;
-    struct passwd *passent, *getpwnam(const char *);
+    struct passwd *passent;
     Str extpath = NULL;
 
     if (name == NULL)
