---
 terms.h         |    4 ++--
 w3mimg/w3mimg.h |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

--- a/terms.h
+++ b/terms.h
@@ -8,7 +8,7 @@ extern int LASTLINE;
 #endif
 
 void clear(void);
-void flush_tty();
+void flush_tty(void);
 void setlinescols(void);
 
 #ifdef USE_MOUSE
@@ -23,7 +23,7 @@ void setlinescols(void);
 #define MOUSE_BTN_UP 3
 #define MOUSE_BTN_RESET -1
 
-void mouse_end();
+void mouse_end(void);
 #endif
 
 #ifdef __CYGWIN__
--- a/w3mimg/w3mimg.h
+++ b/w3mimg/w3mimg.h
@@ -50,7 +50,7 @@ extern w3mimg_op *w3mimg_x11open(void);
 extern w3mimg_op *w3mimg_fbopen(void);
 #endif
 #ifdef USE_W3MIMG_WIN
-extern w3mimg_op *w3mimg_winopen();
+extern w3mimg_op *w3mimg_winopen(void);
 #endif
 
 extern w3mimg_op *w3mimg_open(void);
