---
 libwc/search.c |    1 +
 libwc/search.h |    2 ++
 2 files changed, 3 insertions(+)

--- a/libwc/search.c
+++ b/libwc/search.c
@@ -1,3 +1,4 @@
+#include "search.h"
 
 #include <stdlib.h>
 #include "wc.h"
--- a/libwc/search.h
+++ b/libwc/search.h
@@ -2,6 +2,8 @@
 #ifndef _WC_SEARCH_H
 #define _WC_SEARCH_H
 
+#include "wc_types.h"
+
 extern wc_map  *wc_map_search(wc_uint16 code, wc_map *map, size_t n);
 extern wc_map3 *wc_map3_search(wc_uint16 c1, wc_uint16 c2, wc_map3 *map, size_t n);
 extern wc_map  *wc_map_range_search(wc_uint16 code, wc_map *map, size_t n);
