From: Rene Kita <mail@rkta.de>
Date: Thu Feb 13 09:02:05 CET 2025

---
 acinclude.m4 |  170 ++++++++++++++++++++++++++++++++++++-----------------------
 configure.ac |   46 ++++++++-------
 2 files changed, 129 insertions(+), 87 deletions(-)

--- a/acinclude.m4
+++ b/acinclude.m4
@@ -19,7 +19,7 @@ AC_MSG_CHECKING(if color escape sequence
 AC_ARG_ENABLE(color,
  [  --disable-color         disable color for VT100 terminal],,
  [enable_color="yes"])
-test x"$enable_color" = xyes && AC_DEFINE(USE_COLOR)
+test x"$enable_color" = xyes && AC_DEFINE(USE_COLOR, [], [USE_COLOR])
 AC_MSG_RESULT($enable_color)])
 #
 # ----------------------------------------------------------------
@@ -31,7 +31,8 @@ AC_MSG_CHECKING(if ANSI color escape seq
 AC_ARG_ENABLE(ansi_color,
  [   --disable-ansi-color   disable ANSI color escape sequence],,
  [enable_ansi_color="$enable_color"])
- test x"$enable_ansi_color" = xyes && AC_DEFINE(USE_ANSI_COLOR)
+ test x"$enable_ansi_color" = xyes && AC_DEFINE(USE_ANSI_COLOR, [],
+						[USE_ANSI_COLOR])
  AC_MSG_RESULT($enable_ansi_color)])
 #
 # ----------------------------------------------------------------
@@ -43,7 +44,7 @@ AC_MSG_CHECKING(if background color supp
 AC_ARG_ENABLE(bgcolor,
  [   --disable-bgcolor      disable to set background color],,
  [enable_bgcolor="$enable_color"])
- test x"$enable_bgcolor" = xyes && AC_DEFINE(USE_BG_COLOR)
+ test x"$enable_bgcolor" = xyes && AC_DEFINE(USE_BG_COLOR, [], [USE_BG_COLOR])
 AC_MSG_RESULT($enable_bgcolor)])
 #
 # ----------------------------------------------------------------
@@ -55,7 +56,7 @@ AC_MSG_CHECKING(if popup menu is enabled
 AC_ARG_ENABLE(menu,
  [  --disable-menu          disable popup menu],,
  [enable_menu="yes"])
- test x"$enable_menu" = xyes && AC_DEFINE(USE_MENU)
+ test x"$enable_menu" = xyes && AC_DEFINE(USE_MENU, [], [USE_MENU])
  AC_MSG_RESULT($enable_menu)])
 #
 # ----------------------------------------------------------------
@@ -67,7 +68,7 @@ AC_MSG_CHECKING(if mouse operation enabl
 AC_ARG_ENABLE(mouse,
  [  --disable-mouse         disable mouse operation],,
  [enable_mouse="yes"])
-test x"$enable_mouse" = xyes && AC_DEFINE(USE_MOUSE)
+test x"$enable_mouse" = xyes && AC_DEFINE(USE_MOUSE, [], [USE_MOUSE])
 AC_MSG_RESULT($enable_mouse)])
 #
 # ----------------------------------------------------------------
@@ -79,7 +80,7 @@ AC_MSG_CHECKING(if cookie is enabled)
 AC_ARG_ENABLE(cookie,
  [  --disable-cookie        disable cookie],,
  [enable_cookie="yes"])
-test x"$enable_cookie" = xyes && AC_DEFINE(USE_COOKIE)
+test x"$enable_cookie" = xyes && AC_DEFINE(USE_COOKIE, [], [USE_COOKIE])
 AC_MSG_RESULT($enable_cookie)])
 #
 # ----------------------------------------------------------------
@@ -91,7 +92,7 @@ AC_MSG_CHECKING(if dictionary lookup is
 AC_ARG_ENABLE(dict,
  [  --disable-dict          disable dictionary lookup],,
  [enable_dict="yes"])
- test x"$enable_dict" = xyes && AC_DEFINE(USE_DICT)
+ test x"$enable_dict" = xyes && AC_DEFINE(USE_DICT, [], [USE_DICT])
  AC_MSG_RESULT($enable_dict)])
 #
 # ----------------------------------------------------------------
@@ -103,7 +104,7 @@ AC_MSG_CHECKING(if URL history is enable
 AC_ARG_ENABLE(history,
  [  --disable-history       disable URL history],,
  [enable_history="yes"])
- test x"$enable_history" = xyes && AC_DEFINE(USE_HISTORY)
+ test x"$enable_history" = xyes && AC_DEFINE(USE_HISTORY, [], [USE_HISTORY])
  AC_MSG_RESULT($enable_history)])
 #
 # ----------------------------------------------------------------
@@ -115,7 +116,7 @@ AC_DEFUN([AC_W3M_NNTP],
  AC_ARG_ENABLE(nntp,
   [  --disable-nntp          disable NNTP],,
   [enable_nntp="yes"])
- test x"$enable_nntp" = xyes && AC_DEFINE(USE_NNTP)
+ test x"$enable_nntp" = xyes && AC_DEFINE(USE_NNTP, [], [USE_NNTP])
  AC_MSG_RESULT($enable_nntp)])
 # 
 # ----------------------------------------------------------------
@@ -127,7 +128,7 @@ AC_DEFUN([AC_W3M_GOPHER],
  AC_ARG_ENABLE(gopher,
   [  --disable-gopher        disable Gopher],,
   [enable_gopher="yes"])
- test x"$enable_gopher" = xyes &&  AC_DEFINE(USE_GOPHER)
+ test x"$enable_gopher" = xyes &&  AC_DEFINE(USE_GOPHER, [], [USE_GOPHER])
  AC_MSG_RESULT($enable_gopher)])
 #
 # ----------------------------------------------------------------
@@ -160,7 +161,7 @@ if test x"$enable_m17n" = xno; then
   system_charset='WC_CES_US_ASCII'
   document_charset='WC_CES_US_ASCII'
 else
- AC_DEFINE(USE_M17N)
+ AC_DEFINE(USE_M17N, [], [USE_M17N])
  WCTARGET="libwc/libwc.a"
  WCCFLAGS='-I$(srcdir) -I$(srcdir)/..'
  wcinclude='-I$(srcdir)/libwc'
@@ -180,7 +181,7 @@ else
     if test x"$charset" = xUS-ASCII; then
      charset=UTF-8
     fi
-    AC_DEFINE(USE_UNICODE)
+    AC_DEFINE(USE_UNICODE, [], [USE_UNICODE])
  fi
  AC_MSG_CHECKING(if message l10n)
  AC_ARG_ENABLE(messagel10n,
@@ -250,14 +251,15 @@ else
 fi
 W3M_LANGDEF=`echo $w3m_lang | tr 'a-z' 'A-Z'`
 W3M_LANG=$W3M_LANGDEF
-AC_DEFINE_UNQUOTED(W3M_LANG, $W3M_LANG)
-AC_DEFINE_UNQUOTED(WCTARGET, "$WCTARGET")
-AC_DEFINE_UNQUOTED(WCCFLAGS, "$WCCFLAGS")
+AC_DEFINE_UNQUOTED(W3M_LANG, $W3M_LANG, [Default language])
+AC_DEFINE_UNQUOTED(WCTARGET, "$WCTARGET", [Default WCTARGET])
+AC_DEFINE_UNQUOTED(WCCFLAGS, "$WCCFLAGS", [Default WCCFLAGS])
 CFLAGS="$CFLAGS $wcinclude"
 W3M_LIBS="$W3M_LIBS $wclib"
-AC_DEFINE_UNQUOTED(DISPLAY_CHARSET, $display_charset)
-AC_DEFINE_UNQUOTED(SYSTEM_CHARSET, $system_charset)
-AC_DEFINE_UNQUOTED(DOCUMENT_CHARSET, $document_charset)])
+AC_DEFINE_UNQUOTED(DISPLAY_CHARSET, $display_charset, [Default display charset])
+AC_DEFINE_UNQUOTED(SYSTEM_CHARSET, $system_charset, [Default system charset])
+AC_DEFINE_UNQUOTED(DOCUMENT_CHARSET, $document_charset,
+		   [Default document charset])])
 #
 # ----------------------------------------------------------------
 # AC_W3M_KEYMAP
@@ -279,9 +281,9 @@ AC_DEFUN([AC_W3M_KEYMAP],
  esac
  AC_SUBST(HELP_FILE)
  HELP_FILE=w3mhelp-${enable_keymap}_$w3m_lang.html
- AC_DEFINE_UNQUOTED(HELP_FILE, "$HELP_FILE")
+ AC_DEFINE_UNQUOTED(HELP_FILE, "$HELP_FILE", [Default help file])
  AC_SUBST(KEYBIND)
- AC_DEFINE_UNQUOTED(KEYBIND, $enable_keymap)])
+ AC_DEFINE_UNQUOTED(KEYBIND, $enable_keymap, [Default keymap])])
 #
 # ----------------------------------------------------------------
 # AC_W3M_MIGEMO
@@ -315,11 +317,12 @@ AC_DEFUN([AC_W3M_MIGEMO],
   migemo_command="$migemo_command $f"
  fi
  if test "${with_migemo+set}" = set -a "$with_migemo" != "no"; then
-   AC_DEFINE(USE_MIGEMO)
+   AC_DEFINE(USE_MIGEMO, [], [USE_MIGEMO])
  fi
  AC_MSG_RESULT($migemo_command)
  if test x"$migemo_command" = xno; then migemo_command=""; fi
- AC_DEFINE_UNQUOTED(DEF_MIGEMO_COMMAND, "$migemo_command")])
+ AC_DEFINE_UNQUOTED(DEF_MIGEMO_COMMAND, "$migemo_command",
+		    [Default migemo command])])
 #
 # ----------------------------------------------------------------
 # AC_W3M_EDITOR
@@ -332,7 +335,7 @@ AC_ARG_WITH(editor,
  [  --with-editor=EDITOR    default editor [[/usr/bin/vi]]],
  [w3m_editor="$with_editor"])
 AC_MSG_RESULT($w3m_editor)
-AC_DEFINE_UNQUOTED(DEF_EDITOR, "$w3m_editor")])
+AC_DEFINE_UNQUOTED(DEF_EDITOR, "$w3m_editor", [Default editor])])
 #
 # ----------------------------------------------------------------
 # AC_W3M_MAILER
@@ -345,7 +348,7 @@ AC_ARG_WITH(mailer,
  [  --with-mailer=MAILER    default mailer [[/usr/bin/mail]]],
  [w3m_mailer="$with_mailer"])
 AC_MSG_RESULT($w3m_mailer)
-AC_DEFINE_UNQUOTED(DEF_MAILER, "$w3m_mailer")])
+AC_DEFINE_UNQUOTED(DEF_MAILER, "$w3m_mailer", [Default mailer])])
 #
 # ----------------------------------------------------------------
 # AC_W3M_EXT_BROWSER
@@ -358,7 +361,8 @@ AC_ARG_WITH(browser,
  [  --with-browser=BROWSER  default browser [[/usr/bin/firefox]]],
  [w3m_browser="$with_browser"])
 AC_MSG_RESULT($w3m_browser)
-AC_DEFINE_UNQUOTED(DEF_EXT_BROWSER, "$w3m_browser")])
+AC_DEFINE_UNQUOTED(DEF_EXT_BROWSER, "$w3m_browser",
+		   [Default external browser])])
 #
 # ----------------------------------------------------------------
 # AC_W3M_HELP_CGI
@@ -369,7 +373,7 @@ AC_DEFUN([AC_W3M_HELP_CGI],
  AC_ARG_ENABLE(help_cgi,
   [  --disable-help-cgi      disable help cgi],,
   [enable_help_cgi="yes"])
- test x"$enable_help_cgi" = xyes && AC_DEFINE(USE_HELP_CGI)
+ test x"$enable_help_cgi" = xyes && AC_DEFINE(USE_HELP_CGI, [], [USE_HELP_CGI])
  AC_MSG_RESULT($enable_help_cgi)])
 # 
 # ----------------------------------------------------------------
@@ -381,7 +385,8 @@ AC_DEFUN([AC_W3M_EXTERNAL_URI_LOADER],
  AC_ARG_ENABLE(external_uri_loader,
  [  --disable-external-uri-loader  disable external URI loader],,
  [enable_external_uri_loader="yes"])
- test x"$enable_external_uri_loader" = xyes && AC_DEFINE(USE_EXTERNAL_URI_LOADER)
+ test x"$enable_external_uri_loader" = xyes &&
+	 AC_DEFINE(USE_EXTERNAL_URI_LOADER, [], [USE_EXTERNAL_URI_LOADER])
  AC_MSG_RESULT($enable_external_uri_loader)])
 # 
 # ----------------------------------------------------------------
@@ -394,7 +399,8 @@ AC_DEFUN([AC_W3M_W3MMAILER],
  [  --disable-w3mmailer     disable w3mmailer],,
  [enable_w3mmailer="$enable_external_uri_loader"])
  test x"$enable_external_uri_loader" = xno && enable_w3mmailer=no
- test x"$enable_w3mmailer" = xyes && AC_DEFINE(USE_W3MMAILER)
+ test x"$enable_w3mmailer" = xyes && AC_DEFINE(USE_W3MMAILER, [],
+					       [USE_W3MMAILER])
  AC_MSG_RESULT($enable_w3mmailer)])
 #
 # ----------------------------------------------------------------
@@ -522,12 +528,13 @@ if test x"$with_ssl" != xno; then
 	[$SSL_LIBS -lcrypto])
 
   if test x"$w3m_ssl" = xfound; then
-    AC_DEFINE(USE_SSL)
+    AC_DEFINE(USE_SSL, [], [USE_SSL])
     AC_MSG_CHECKING(if SSL certificate verification is enabled)
     AC_ARG_ENABLE(sslverify,
       [  --disable-sslverify     disable SSL certificate verification],,
       [enable_sslverify="yes"])
-    test x"$enable_sslverify" = xyes && AC_DEFINE(USE_SSL_VERIFY)
+    test x"$enable_sslverify" = xyes && AC_DEFINE(USE_SSL_VERIFY, [],
+						  [USE_SSL_VERIFY])
     AC_MSG_RESULT($enable_sslverify)
   fi
 fi
@@ -553,14 +560,14 @@ if test x"$enable_sslverify" = xyes; the
   fi
   AC_MSG_RESULT($w3m_cafile)
 fi
-AC_DEFINE_UNQUOTED(DEF_CAFILE, "$w3m_cafile")
+AC_DEFINE_UNQUOTED(DEF_CAFILE, "$w3m_cafile", [Default CA file])
 AC_SUBST(USE_DIGEST_AUTH)
 AC_MSG_CHECKING(if digest auth is enabled)
 AC_ARG_ENABLE(digest_auth,
  [  --disable-digest-auth   disable digest auth],,
  [enable_digest_auth="yes"])
 if test x"$enable_digest_auth" = xyes -a x"$w3m_ssl" = xfound; then
-  AC_DEFINE(USE_DIGEST_AUTH)
+  AC_DEFINE(USE_DIGEST_AUTH, [], [USE_DIGEST_AUTH])
 else
   enable_digest_auth="no"
 fi
@@ -580,7 +587,7 @@ AC_DEFUN([AC_W3M_ALARM],
  if test x"$enable_alarm" = xyes; then
    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <unistd.h>
 #include <signal.h>]], [[int sa = SIGALRM;
-     void (*a) = alarm;]])],[AC_DEFINE(USE_ALARM)],[])
+     void (*a) = alarm;]])],[AC_DEFINE(USE_ALARM, [], [USE_ALARM])],[])
  fi])
 #
 # ----------------------------------------------------------------
@@ -656,14 +663,15 @@ AC_DEFUN([AC_W3M_IMAGE],
       x11) x11=yes;;
       fb)  fb=yes;;
       fb+s) fb=yes
-           AC_DEFINE(W3MIMGDISPLAY_SETUID)
+           AC_DEFINE(W3MIMGDISPLAY_SETUID, [], [W3MIMGDISPLAY_SETUID])
            INSTALL_W3MIMGDISPLAY='${INSTALL} -o root -m 4755 -s'
-           AC_DEFINE(INSTALL_W3MIMGDISPLAY, $INSTALL_W3MIMGDISPLAY);;
+           AC_DEFINE(INSTALL_W3MIMGDISPLAY, $INSTALL_W3MIMGDISPLAY,
+		     [INSTALL_W3MIMGDISPLAY]);;
       win) win=yes;;
     esac
   done
   enable_image=yes
-  AC_DEFINE(USE_IMAGE)
+  AC_DEFINE(USE_IMAGE, [], [USE_IMAGE])
   AC_MSG_CHECKING(image library)
   AC_ARG_WITH(imagelib,
    [  --with-imagelib[[=LIBS]]  image library [[guessed]]
@@ -737,32 +745,32 @@ AC_DEFUN([AC_W3M_IMAGE],
   fi
   if test x"$x11" = xyes; then
    if test x"$have_imlib2" = xyes; then
-     AC_DEFINE(USE_W3MIMG_X11)
+     AC_DEFINE(USE_W3MIMG_X11, [], [USE_W3MIMG_X11])
      IMGOBJS="$IMGOBJS x11/x11_w3mimg.o"
      IMGTARGETS="x11"    
-     AC_DEFINE(USE_IMLIB2)
+     AC_DEFINE(USE_IMLIB2, [], [USE_IMLIB2])
      IMGX11CFLAGS="`${PKG_CONFIG} --cflags imlib2`"
      IMGX11LDFLAGS="-lX11 `${PKG_CONFIG} --libs imlib2`"
    elif test x"$have_gtk2" = xyes; then
-     AC_DEFINE(USE_W3MIMG_X11)
+     AC_DEFINE(USE_W3MIMG_X11, [], [USE_W3MIMG_X11])
      IMGOBJS="$IMGOBJS x11/x11_w3mimg.o"
      IMGTARGETS="x11"    
-     AC_DEFINE(USE_GDKPIXBUF)
-     AC_DEFINE(USE_GTK2)
+     AC_DEFINE(USE_GDKPIXBUF, [], [USE_GDKPIXBUF])
+     AC_DEFINE(USE_GTK2, [], [USE_GTK2])
      IMGX11CFLAGS="`${PKG_CONFIG} --cflags gdk-pixbuf-2.0 gdk-pixbuf-xlib-2.0`"
      IMGX11LDFLAGS="-lX11 `${PKG_CONFIG} --libs gdk-pixbuf-2.0 gdk-pixbuf-xlib-2.0`"
    elif test x"$have_gdkpixbuf" = xyes; then
-     AC_DEFINE(USE_W3MIMG_X11)
+     AC_DEFINE(USE_W3MIMG_X11, [], [USE_W3MIMG_X11])
      IMGOBJS="$IMGOBJS x11/x11_w3mimg.o"
      IMGTARGETS="x11"    
-     AC_DEFINE(USE_GDKPIXBUF)
+     AC_DEFINE(USE_GDKPIXBUF, [], [USE_GDKPIXBUF])
      IMGX11CFLAGS="`${GDKPIXBUF_CONFIG} --cflags`"
      IMGX11LDFLAGS="`${GDKPIXBUF_CONFIG} --libs` -lgdk_pixbuf_xlib"
    elif test x"$have_imlib" = xyes; then
-     AC_DEFINE(USE_W3MIMG_X11)
+     AC_DEFINE(USE_W3MIMG_X11, [], [USE_W3MIMG_X11])
      IMGOBJS="$IMGOBJS x11/x11_w3mimg.o"
      IMGTARGETS="x11"    
-     AC_DEFINE(USE_IMLIB)
+     AC_DEFINE(USE_IMLIB, [], [USE_IMLIB])
      IMGX11CFLAGS="`${IMLIB_CONFIG} --cflags`"
      IMGX11LDFLAGS="`${IMLIB_CONFIG} --libs`"
      IMGTARGETS="x11"    
@@ -772,26 +780,26 @@ AC_DEFUN([AC_W3M_IMAGE],
   fi
   if test x"$fb" = xyes; then
    if test x"$have_imlib2" = xyes; then
-     AC_DEFINE(USE_W3MIMG_FB)
+     AC_DEFINE(USE_W3MIMG_FB, [], [USE_W3MIMG_FB])
      IMGOBJS="$IMGOBJS fb/fb_w3mimg.o fb/fb.o fb/fb_img.o"
      IMGTARGETS="${IMGTARGETS} fb"
-     AC_DEFINE(USE_IMLIB2)
+     AC_DEFINE(USE_IMLIB2, [], [USE_IMLIB2])
      IMGOBJS="$IMGOBJS fb/fb_w3mimg.o fb/fb.o fb/fb_img.o"
      IMGFBCFLAGS="`${PKG_CONFIG} --cflags imlib2`"
      IMGFBLDFLAGS="`${PKG_CONFIG} --libs imlib2`"
    elif test x"$have_gtk2" = xyes; then
-     AC_DEFINE(USE_W3MIMG_FB)
+     AC_DEFINE(USE_W3MIMG_FB, [], [USE_W3MIMG_FB])
      IMGOBJS="$IMGOBJS fb/fb_w3mimg.o fb/fb.o fb/fb_img.o"
      IMGTARGETS="${IMGTARGETS} fb"
-     AC_DEFINE(USE_GDKPIXBUF)
-     AC_DEFINE(USE_GTK2)
+     AC_DEFINE(USE_GDKPIXBUF, [], [USE_GDKPIXBUF])
+     AC_DEFINE(USE_GTK2, [], [USE_GTK2])
      IMGFBCFLAGS="`${PKG_CONFIG} --cflags gdk-pixbuf-2.0`"
      IMGFBLDFLAGS="`${PKG_CONFIG} --libs gdk-pixbuf-2.0`"
    elif test x"$have_gdkpixbuf" = xyes; then
-     AC_DEFINE(USE_W3MIMG_FB)
+     AC_DEFINE(USE_W3MIMG_FB, [], [USE_W3MIMG_FB])
      IMGOBJS="$IMGOBJS fb/fb_w3mimg.o fb/fb.o fb/fb_img.o"
      IMGTARGETS="${IMGTARGETS} fb"
-     AC_DEFINE(USE_GDKPIXBUF)
+     AC_DEFINE(USE_GDKPIXBUF, [], [USE_GDKPIXBUF])
      IMGFBCFLAGS="`${GDKPIXBUF_CONFIG} --cflags`"
      IMGFBLDFLAGS="`${GDKPIXBUF_CONFIG} --libs`"
    else
@@ -799,22 +807,22 @@ AC_DEFUN([AC_W3M_IMAGE],
    fi
   fi
   if test x"$win" = xyes; then
-    AC_DEFINE(USE_W3MIMG_WIN)
+    AC_DEFINE(USE_W3MIMG_WIN, [], [USE_W3MIMG_WIN])
     IMGOBJS="$IMGOBJS win/win_w3mimg.o"
     IMGTARGETS="${IMGTARGETS} win"
     IMGWINCFLAGS="-I/usr/include/w32api"
     IMGWINLDFLAGS="-lgdiplus -lgdi32 -luser32"
     IMGLINK='$(CXX)'
   fi
-  AC_DEFINE(IMGTARGETS, "$IMGTARGETS")
-  AC_DEFINE(IMGOBJS, "$IMGOBJS")
-  AC_DEFINE(IMGX11CFLAGS, "$IMGX11CFLAGS")
-  AC_DEFINE(IMGX11LDFLAGS, "$IMGX11LDFLAGS")
-  AC_DEFINE(IMGFBCFLAGS, "$IMGFBCFLAGS")
-  AC_DEFINE(IMGFBLDFLAGS, "$IMGFBLDFLAGS")
-  AC_DEFINE(IMGLINK, "$IMGLINK")
-  AC_DEFINE(IMGWINCFLAGS, "$IMGWINCFLAGS")
-  AC_DEFINE(IMGWINLDFLAGS, "$IMGWINLDFLAGS")
+  AC_DEFINE(IMGTARGETS, "$IMGTARGETS", [IMGTARGETS])
+  AC_DEFINE(IMGOBJS, "$IMGOBJS", [IMGOBJS])
+  AC_DEFINE(IMGX11CFLAGS, "$IMGX11CFLAGS", [IMGX11CFLAGS])
+  AC_DEFINE(IMGX11LDFLAGS, "$IMGX11LDFLAGS", [IMGX11LDFLAGS])
+  AC_DEFINE(IMGFBCFLAGS, "$IMGFBCFLAGS", [IMGFBCFLAGS])
+  AC_DEFINE(IMGFBLDFLAGS, "$IMGFBLDFLAGS", [IMGFBLDFLAGS])
+  AC_DEFINE(IMGLINK, "$IMGLINK", [IMGLINK])
+  AC_DEFINE(IMGWINCFLAGS, "$IMGWINCFLAGS", [IMGWINCFLAGS])
+  AC_DEFINE(IMGWINLDFLAGS, "$IMGWINLDFLAGS", [IMGWINLDFLAGS])
  fi])
 # ----------------------------------------------------------------
 # AC_W3M_XFACE
@@ -825,7 +833,7 @@ AC_DEFUN([AC_W3M_XFACE],
  AC_ARG_ENABLE(xface,
   [   --disable-xface        disable X-Face support],,
   [enable_xface="$enable_image"])
- test x"$enable_xface" = xyes && AC_DEFINE(USE_XFACE)
+ test x"$enable_xface" = xyes && AC_DEFINE(USE_XFACE, [], [USE_XFACE])
  AC_MSG_RESULT($enable_xface)
  AC_CHECK_PROG(uncompface, uncompface, "yes", "no")
  test "$uncompface" = "no" && AC_MSG_WARN([uncompface is not installed.])
@@ -877,7 +885,7 @@ if test x"$enable_ipv6" = xyes; then
       enable_ipv6="no"])
  fi
  if test x"$enable_ipv6" = xyes; then
-    AC_DEFINE(INET6)
+    AC_DEFINE(INET6, [], [INET6])
  fi
 fi])
 #
@@ -885,9 +893,39 @@ fi])
 # AC_W3M_SIGSETJMP
 # ----------------------------------------------------------------
 AC_DEFUN([AC_W3M_SIGSETJMP],
-[AC_SUBST(HAVE_SIGSETJMP)
+[AC_SUBST(HAVE_SIGSETJMP, [], [HAVE_SIGSETJMP])
 AC_MSG_CHECKING(for sigsetjmp)
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <setjmp.h>
  #include <stdlib.h>]], [[ jmp_buf env;
-   if (sigsetjmp(env, 1) != 0) { exit(0); } siglongjmp(env, 1);]])],[have_sigsetjmp="yes"; AC_DEFINE(HAVE_SIGSETJMP)],[have_sigsetjmp="no"])
+   if (sigsetjmp(env, 1) != 0) { exit(0); } siglongjmp(env, 1);]])],[have_sigsetjmp="yes"; AC_DEFINE(HAVE_SIGSETJMP, [], [HAVE_SIGSETJMP])],[have_sigsetjmp="no"])
 AC_MSG_RESULT($have_sigsetjmp)])
+dnl
+dnl --------------------------------------------------------------
+dnl autoconf macros from external sources
+dnl --------------------------------------------------------------
+# codeset.m4
+dnl # serial 5 (gettext-0.18.2)
+dnl Copyright (C) 2000-2002, 2006, 2008-2014, 2016, 2019-2025 Free Software
+dnl Foundation, Inc.
+dnl This file is free software; the Free Software Foundation
+dnl gives unlimited permission to copy and/or distribute it,
+dnl with or without modifications, as long as this notice is preserved.
+dnl This file is offered as-is, without any warranty.
+
+dnl From Bruno Haible.
+
+AC_DEFUN([AM_LANGINFO_CODESET],
+[
+  AC_CACHE_CHECK([for nl_langinfo and CODESET], [am_cv_langinfo_codeset],
+    [AC_LINK_IFELSE(
+       [AC_LANG_PROGRAM(
+          [[#include <langinfo.h>]],
+          [[char* cs = nl_langinfo(CODESET); return !cs;]])],
+       [am_cv_langinfo_codeset=yes],
+       [am_cv_langinfo_codeset=no])
+    ])
+  if test $am_cv_langinfo_codeset = yes; then
+    AC_DEFINE([HAVE_LANGINFO_CODESET], [1],
+      [Define if you have <langinfo.h> and nl_langinfo(CODESET).])
+  fi
+])
--- a/configure.ac
+++ b/configure.ac
@@ -3,14 +3,14 @@ dnl Process this file with autoconf to p
 AC_INIT([w3m],[0.5.3])
 PACKAGE=w3m
 VERSION=0.5.3
-AC_DEFINE_UNQUOTED(PACKAGE, "$PACKAGE")
-AC_DEFINE_UNQUOTED(VERSION, "$VERSION")
+AC_DEFINE_UNQUOTED(PACKAGE, "$PACKAGE", [PACKAGE])
+AC_DEFINE_UNQUOTED(VERSION, "$VERSION", [VERSION])
 AC_SUBST(PACKAGE)
 AC_SUBST(VERSION)
 
 AC_SUBST(W3M)
 W3M=w3m
-AC_DEFINE(W3M, w3m)
+AC_DEFINE(W3M, w3m, [w3m])
 
 AC_SUBST(W3M_LANG)
 W3M_LANG=${LC_ALL:-$LANG}
@@ -39,7 +39,7 @@ AUXBIN_TARGETS=
 AC_SUBST(W3M_TARGET)
 AC_SUBST(W3M_LIBS)
 W3M_TARGET=$target
-AC_DEFINE_UNQUOTED(W3M_TARGET, $W3M_TARGET)
+AC_DEFINE_UNQUOTED(W3M_TARGET, $W3M_TARGET, [W3M_TARGET])
 
 AC_W3M_M17N
 if test x$enable_messagel10n = xno; then
@@ -95,13 +95,13 @@ dnl Checks for libraries.
 AC_CHECK_FUNC(sqrt,,[AC_CHECK_LIB(m, sqrt)])
 if test x"$enable_mouse" = xyes; then
  AC_SUBST(USE_W3M)
- AC_CHECK_LIB(gpm, Gpm_Open, [AC_DEFINE(USE_GPM)
+ AC_CHECK_LIB(gpm, Gpm_Open, [AC_DEFINE(USE_GPM, [], [USE_GPM])
    		W3M_LIBS="$W3M_LIBS -lgpm"])
  AC_SUBST(USE_SYSMOUSE)
  AC_COMPILE_IFELSE([AC_LANG_SOURCE([#include <sys/consio.h>
-   int test = MOUSE_GETINFO;])], [AC_DEFINE(USE_SYSMOUSE)],
+   int test = MOUSE_GETINFO;])], [AC_DEFINE(USE_SYSMOUSE, [], [USE_SYSMOUSE])],
    [AC_COMPILE_IFELSE([AC_LANG_SOURCE([#include <machine/console.h>
-      MOUSE_GETINFO], [AC_DEFINE(USE_SYSMOUSE)])])])
+      MOUSE_GETINFO], [AC_DEFINE(USE_SYSMOUSE, [], [USE_SYSMOUSE])])])])
 fi
 AC_SUBST(AUXBIN_TARGETS)
 AC_W3M_TERMLIB
@@ -109,29 +109,33 @@ AC_W3M_GC
 AC_CHECK_LIB(z, zError, [AUXBIN_TARGETS="$AUXBIN_TARGETS inflate\$(EXT)"])
 
 AC_CHECK_TYPE([socklen_t],
-	 [AC_DEFINE(HAVE_SOCKLEN_T)], ,
+	 [AC_DEFINE(HAVE_SOCKLEN_T, [], [HAVE_SOCKLEN_T])], ,
 	 [#include <sys/types.h>
 	  #include <sys/socket.h>])
 
-AC_DEFINE(AUXBIN_TARGETS, "$AUXBIN_TARGETS")
+AC_DEFINE(AUXBIN_TARGETS, "$AUXBIN_TARGETS", [Description])
 
 AC_SUBST(USE_BINMODE_STREAM)
 case "$host_os" in
   *cygwin*|*os2-emx*|*mingw*)
-    AC_DEFINE(USE_BINMODE_STREAM)
+    AC_DEFINE(USE_BINMODE_STREAM, [], [USE_BINMODE_STREAM])
     ;;
 esac
 
 AC_HEADER_DIRENT
 AC_HEADER_SYS_WAIT
-AC_CHECK_HEADER(termios.h, [AC_DEFINE(HAVE_TERMIOS_H)],
-	 [AC_CHECK_HEADER(termio.h, [AC_DEFINE(HAVE_TERMIO_H)],
-	  [AC_CHECK_HEADER(sgtty.h, [AC_DEFINE(HAVE_SGTTY_H)])])])
-AC_CHECK_HEADER([float.h], [AC_DEFINE(HAVE_FLOAT_H)])
-AC_CHECK_HEADER([sys/select.h], [AC_DEFINE(HAVE_SYS_SELECT_H)])
-AC_CHECK_HEADER([locale.h], [AC_DEFINE(HAVE_LOCALE_H)])
-AC_CHECK_HEADER([stdint.h], [AC_DEFINE(HAVE_STDINT_H)])
-AC_CHECK_HEADER([inttypes.h], [AC_DEFINE(HAVE_INTTYPES_H)])
+AC_CHECK_HEADER(termios.h, [AC_DEFINE(HAVE_TERMIOS_H, [], [HAVE_TERMIOS_H])],
+	 [AC_CHECK_HEADER(termio.h, [AC_DEFINE(HAVE_TERMIO_H, [],
+			  [HAVE_TERMIO_H])],
+	  [AC_CHECK_HEADER(sgtty.h, [AC_DEFINE(HAVE_SGTTY_H, [],
+			   [HAVE_SGTTY_H])])])])
+AC_CHECK_HEADER([float.h], [AC_DEFINE(HAVE_FLOAT_H, [], [HAVE_FLOAT_H])])
+AC_CHECK_HEADER([sys/select.h], [AC_DEFINE(HAVE_SYS_SELECT_H, [],
+		[HAVE_SYS_SELECT_H])])
+AC_CHECK_HEADER([locale.h], [AC_DEFINE(HAVE_LOCALE_H, [], [HAVE_LOCALE_H])])
+AC_CHECK_HEADER([stdint.h], [AC_DEFINE(HAVE_STDINT_H, [], [HAVE_STDINT_H])])
+AC_CHECK_HEADER([inttypes.h], [AC_DEFINE(HAVE_INTTYPES_H, [],
+		[HAVE_INTTYPES_H])])
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_CHECK_SIZEOF(long long, 0)
@@ -151,13 +155,13 @@ AC_W3M_SIGSETJMP
 
 AC_SUBST(HELP_DIR)
 HELP_DIR="$datadir/$PACKAGE"
-AC_DEFINE(HELP_DIR, "$HELP_DIR")
+AC_DEFINE(HELP_DIR, "$HELP_DIR", [Description])
 AC_SUBST(RC_DIR)
 RC_DIR="~/.$PACKAGE"
-AC_DEFINE(RC_DIR, "$RC_DIR")
+AC_DEFINE(RC_DIR, "$RC_DIR", [Description])
 AC_SUBST(DOCDIRS)
 DOCDIRS="doc:en_English doc-jp:ja_Japanese doc-de:de_German"
-AC_DEFINE_UNQUOTED(DOCDIRS, "$DOCDIRS")
+AC_DEFINE_UNQUOTED(DOCDIRS, "$DOCDIRS", [DOCDIRS])
 
 AC_W3M_VERSION
 AC_CONFIG_FILES([Makefile posubst po/Makefile.in \
