--- a/istream.c
+++ b/istream.c
@@ -322,15 +322,34 @@ ISfileno(InputStream stream)
 }
 
 #ifdef USE_SSL
-static Str accept_this_site;
+static TextList *known_hosts;
+
+static Str
+X509_fingerprint(X509 *x)
+{
+    Str fp = Strnew();
+    const EVP_MD *digest;
+    unsigned n;
+    unsigned char md[EVP_MAX_MD_SIZE];
+
+    digest = EVP_get_digestbyname("sha1");
+    X509_digest(x, digest, md, &n);
+    for(int i = 0; i < n - 1; i++)
+	Strcat(fp, Sprintf("%02x:", md[i]));
+    Strcat(fp, Sprintf("%02x", md[n - 1]));
+
+    return fp;
+}
 
 void
-ssl_accept_this_site(char *hostname)
+ssl_accept_this_site(char *hostname, X509 *x)
 {
-    if (hostname)
-	accept_this_site = Strnew_charp(hostname);
-    else
-	accept_this_site = NULL;
+    const char *hh;
+
+    if (!hostname) return;
+
+    hh = Strnew_m_charp(hostname, " ", X509_fingerprint(x)->ptr, NULL)->ptr;
+    pushText(known_hosts, hh);
 }
 
 static int
@@ -469,6 +488,36 @@ ssl_check_cert_ident(X509 * x, char *hos
     return ret;
 }
 
+/* Return true if the host's cert was manually accepted before */
+static int
+accept_this_site(const char *hostname, X509 *x)
+{
+    Str fp;
+    TextListItem *host;
+    size_t n;
+
+    if (!known_hosts)
+	known_hosts = newTextList();
+
+    if (!(host = known_hosts->last))
+	return 0;
+
+    n = strlen(hostname);
+    fp = X509_fingerprint(x);
+    if (!strncmp(host->ptr, hostname, n)
+	&& !strncmp(host->ptr + n + 1, fp->ptr, fp->length))
+	return 1;
+
+    for (host = host->prev; host; host = host->prev) {
+	if (!strncmp(host->ptr, hostname, n)
+	    && !strncmp(host->ptr + n + 1, fp->ptr, fp->length)) {
+	    delText(known_hosts, host);
+	    return 1;
+	}
+    }
+    return 0;
+}
+
 Str
 ssl_get_certificate(SSL * ssl, char *hostname)
 {
@@ -485,10 +534,10 @@ ssl_get_certificate(SSL * ssl, char *hos
 
     if (ssl == NULL)
 	return NULL;
+
     x = SSL_get_peer_certificate(ssl);
     if (x == NULL) {
-	if (accept_this_site
-	    && strcasecmp(accept_this_site->ptr, hostname) == 0)
+	if (accept_this_site(hostname, x))
 	    ans = "y";
 	else {
 	    /* FIXME: gettextize? */
@@ -509,7 +558,7 @@ ssl_get_certificate(SSL * ssl, char *hos
 	}
 	if (amsg)
 	    disp_err_message(amsg->ptr, FALSE);
-	ssl_accept_this_site(hostname);
+	ssl_accept_this_site(hostname, x);
 	/* FIXME: gettextize? */
 	s = amsg ? amsg : Strnew_charp("valid certificate");
 	return s;
@@ -524,34 +573,33 @@ ssl_get_certificate(SSL * ssl, char *hos
 	if ((verr = SSL_get_verify_result(ssl))
 	    != X509_V_OK) {
 	    const char *em = X509_verify_cert_error_string(verr);
-	    if (accept_this_site
-		&& strcasecmp(accept_this_site->ptr, hostname) == 0)
+	    if (accept_this_site(hostname, x))
 		ans = "y";
 	    else {
 		/* FIXME: gettextize? */
 		emsg = Sprintf("%s: accept? (y/n)", em);
 		ans = inputAnswer(emsg->ptr);
-	    }
-	    if (ans && TOLOWER(*ans) == 'y') {
-		/* FIXME: gettextize? */
-		amsg = Sprintf("Accept unsecure SSL session: "
-			       "unverified: %s", em);
-	    }
-	    else {
-		/* FIXME: gettextize? */
-		char *e =
-		    Sprintf("This SSL session was rejected: %s", em)->ptr;
-		disp_err_message(e, FALSE);
-		free_ssl_ctx();
-		return NULL;
+		if (ans && TOLOWER(*ans) == 'y') {
+		    /* FIXME: gettextize? */
+		    amsg = Sprintf("Accept unsecure SSL session: "
+				   "unverified: %s", em);
+		}
+		else {
+		    /* FIXME: gettextize? */
+		    char *e =
+			Sprintf("This SSL session was rejected: %s", em)->ptr;
+		    disp_err_message(e, FALSE);
+		    free_ssl_ctx();
+		    return NULL;
+		}
 	    }
 	}
+	ssl_accept_this_site(hostname, x);
     }
 #endif
     emsg = ssl_check_cert_ident(x, hostname);
     if (emsg != NULL) {
-	if (accept_this_site
-	    && strcasecmp(accept_this_site->ptr, hostname) == 0)
+	if (accept_this_site(hostname, x))
 	    ans = "y";
 	else {
 	    Str ep = Strdup(emsg);
@@ -573,10 +621,10 @@ ssl_get_certificate(SSL * ssl, char *hos
 	    free_ssl_ctx();
 	    return NULL;
 	}
+	ssl_accept_this_site(hostname, x);
     }
     if (amsg)
 	disp_err_message(amsg->ptr, FALSE);
-    ssl_accept_this_site(hostname);
     /* FIXME: gettextize? */
     s = amsg ? amsg : Strnew_charp("valid certificate");
     Strcat_charp(s, "\n");
