From f4d30368a39f52e8eabc32a7af213efa7e48c747 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Sun, 19 Nov 2023 10:33:43 +0100
Subject: [PATCH 1/2] Label #endif/#else that is far away from #ifdef

---
 etc.c |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/etc.c
+++ b/etc.c
@@ -317,6 +317,7 @@ checkType(Str s, Lineprop **oprop, Linec
 	    Strcat_charp_n(s, sp, (int)(str - sp));
 	}
     }
+
     if (!do_copy) {
 	for (; str < endp && IS_ASCII(*str); str++) {
 	    *(prop++) = PE_NORMAL | (IS_CNTRL(*str) ? PC_CTRL : PC_ASCII);
@@ -404,7 +405,7 @@ checkType(Str s, Lineprop **oprop, Linec
 			str += 2;
 		    }
 		}
-#endif
+#endif	/* USE_M17N */
 		else {
 		    if (s->length) {
 #ifdef USE_M17N
@@ -428,7 +429,7 @@ checkType(Str s, Lineprop **oprop, Linec
 				plen = *(--plens);
 			    str++;
 			}
-#else
+#else	/* USE_M17N */
 			if (*(str - 1) == *(str + 1)) {
 			    *(prop - 1) |= PE_BOLD;
 			    str += 2;
@@ -442,7 +443,7 @@ checkType(Str s, Lineprop **oprop, Linec
 #endif
 			    str++;
 			}
-#endif
+#endif	/* USE_M17N */
 		    }
 		    else {
 			str++;
@@ -457,6 +458,7 @@ checkType(Str s, Lineprop **oprop, Linec
 		bs = memchr(str, '\b', endp - str);
 #endif
 	}
+
 #ifdef USE_ANSI_COLOR
 	if (es != NULL) {
 	    if (str == es) {
@@ -501,7 +503,7 @@ checkType(Str s, Lineprop **oprop, Linec
 	    str += plen;
 	}
 	else
-#endif
+#endif	/* USE_M17N */
 	{
 	    if (do_copy)
 		Strcat_char(s, (char)*str);
