From: Rene Kita <mail@rkta.de>
Date: Tue Aug 12 17:29:13 CEST 2025

---
 main.c  |    2 ++
 proto.h |    1 -
 2 files changed, 2 insertions(+), 1 deletion(-)

--- a/main.c
+++ b/main.c
@@ -130,6 +130,8 @@ static int check_target = TRUE;
 #define PREC_LIMIT 10000
 static int searchKeyNum(void);
 
+static void intTrap(SIGNAL_ARG);
+
 #define help() fusage(stdout, 0)
 #define usage() fusage(stderr, 1)
 
--- a/proto.h
+++ b/proto.h
@@ -9,7 +9,6 @@
 #define W3M_PROTO_H__
 extern void nulcmd(void);
 extern void pushEvent(int cmd, void *data);
-extern void intTrap(SIGNAL_ARG);
 extern void pgFore(void);
 extern void pgBack(void);
 extern void hpgFore(void);
