--- a/file.c
+++ b/file.c
@@ -1513,7 +1513,7 @@ static void
 getAuthCookie(struct http_auth *hauth, char *auth_header,
 	      TextList *extra_header, ParsedURL *pu, HRequest *hr,
 	      FormList *request,
-	      volatile Str *uname, volatile Str *pwd)
+	      Str *uname, Str *pwd)
 {
     Str ss = NULL;
     Str tmp;
@@ -1695,28 +1695,28 @@ getLinkNumberStr(int correction)
  */
 #define DO_EXTERNAL ((Buffer *(*)(URLFile *, Buffer *))doExternal)
 Buffer *
-loadGeneralFile(char *path, ParsedURL *volatile current, char *referer,
-		int flag, FormList *volatile request)
+loadGeneralFile(char *path, ParsedURL *current, char *referer,
+		int flag, FormList *request)
 {
-    URLFile f, *volatile of = NULL;
+    URLFile f, *of = NULL;
     ParsedURL pu;
     Buffer *b = NULL;
-    Buffer *(*volatile proc)(URLFile *, Buffer *) = loadBuffer;
-    char *volatile tpath;
-    char *volatile t = "text/plain", *p, *volatile real_type = NULL;
-    Buffer *volatile t_buf = NULL;
-    int volatile searchHeader = SearchHeader;
-    int volatile searchHeader_through = TRUE;
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    Buffer *(*proc)(URLFile *, Buffer *) = loadBuffer;
+    char *tpath;
+    char *t = "text/plain", *p, *real_type = NULL;
+    Buffer *t_buf = NULL;
+    int searchHeader = SearchHeader;
+    int searchHeader_through = TRUE;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
     TextList *extra_header = newTextList();
-    volatile Str uname = NULL;
-    volatile Str pwd = NULL;
-    volatile Str realm = NULL;
-    int volatile add_auth_cookie_flag;
+    Str uname = NULL;
+    Str pwd = NULL;
+    Str realm = NULL;
+    int add_auth_cookie_flag;
     unsigned char status = HTST_NORMAL;
     URLOption url_option;
     Str tmp;
-    Str volatile page = NULL;
+    Str page = NULL;
 #ifdef USE_GOPHER
     int gopher_download = FALSE;
 #endif
@@ -1724,7 +1724,7 @@ loadGeneralFile(char *path, ParsedURL *v
     wc_ces charset = WC_CES_US_ASCII;
 #endif
     HRequest hr;
-    ParsedURL *volatile auth_pu;
+    ParsedURL *auth_pu;
 
     tpath = path;
     prevtrap = NULL;
@@ -7188,14 +7188,14 @@ loadHTMLstream(URLFile *f, Buffer *newBu
     Str lineBuf2 = Strnew();
 #ifdef USE_M17N
     wc_ces charset = WC_CES_US_ASCII;
-    wc_ces volatile doc_charset = DocumentCharset;
+    wc_ces doc_charset = DocumentCharset;
 #endif
     struct html_feed_environ htmlenv1;
     struct readbuffer obuf;
 #ifdef USE_IMAGE
-    int volatile image_flag;
+    int image_flag;
 #endif
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
 
 #ifdef USE_M17N
     if (fmInitialized && graph_ok()) {
@@ -7359,7 +7359,7 @@ Buffer *
 loadHTMLString(Str page)
 {
     URLFile f;
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
     Buffer *newBuf;
 
     init_stream(&f, SCM_LOCAL, newStrStream(page));
@@ -7406,11 +7406,11 @@ Str
 loadGopherDir0(URLFile *uf, ParsedURL *pu)
 #endif
 {
-    Str volatile tmp;
+    Str tmp;
     Str lbuf, name, file, host, port, type;
-    char *volatile p, *volatile q;
+    char *p, *q;
     int link, pre;
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
 #ifdef USE_M17N
     wc_ces doc_charset = DocumentCharset;
 #endif
@@ -7533,7 +7533,7 @@ loadGopherSearch0(URLFile *uf, ParsedURL
 #endif
 {
     Str tmp;
-    char *volatile p, *volatile q;
+    char *p, *q;
 #ifdef USE_M17N
     wc_ces doc_charset = DocumentCharset;
 #endif
@@ -7558,15 +7558,15 @@ loadGopherSearch0(URLFile *uf, ParsedURL
  * loadBuffer: read file and make new buffer
  */
 Buffer *
-loadBuffer(URLFile *uf, Buffer *volatile newBuf)
+loadBuffer(URLFile *uf, Buffer *newBuf)
 {
-    FILE *volatile src = NULL;
+    FILE *src = NULL;
 #ifdef USE_M17N
     wc_ces charset = WC_CES_US_ASCII;
-    wc_ces volatile doc_charset = DocumentCharset;
+    wc_ces doc_charset = DocumentCharset;
 #endif
     Str lineBuf2;
-    volatile char pre_lbuf = '\0';
+    char pre_lbuf = '\0';
     int nlines;
     Str tmpf;
     clen_t linelen = 0, trbyte = 0;
@@ -7574,7 +7574,7 @@ loadBuffer(URLFile *uf, Buffer *volatile
 #ifdef USE_ANSI_COLOR
     Linecolor *colorBuffer = NULL;
 #endif
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
 
     if (newBuf == NULL)
 	newBuf = newBuffer(INIT_BUFFER_WIDTH);
@@ -7663,7 +7663,7 @@ loadImageBuffer(URLFile *uf, Buffer *new
     Str tmp, tmpf;
     FILE *src = NULL;
     URLFile f;
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
     struct stat st;
     const ParsedURL *pu = newBuf ? &newBuf->currentURL : NULL;
 
@@ -7973,26 +7973,26 @@ openGeneralPagerBuffer(InputStream strea
 Line *
 getNextPage(Buffer *buf, int plen)
 {
-    Line *volatile top = buf->topLine, *volatile last = buf->lastLine,
-	*volatile cur = buf->currentLine;
+    Line *top = buf->topLine, *last = buf->lastLine,
+	*cur = buf->currentLine;
     int i;
-    int volatile nlines = 0;
+    int nlines = 0;
     clen_t linelen = 0, trbyte = buf->trbyte;
     Str lineBuf2;
-    char volatile pre_lbuf = '\0';
+    char pre_lbuf = '\0';
     URLFile uf;
 #ifdef USE_M17N
     wc_ces charset;
-    wc_ces volatile doc_charset = DocumentCharset;
+    wc_ces doc_charset = DocumentCharset;
     wc_uint8 old_auto_detect = WcOption.auto_detect;
 #endif
-    int volatile squeeze_flag = FALSE;
+    int squeeze_flag = FALSE;
     Lineprop *propBuffer = NULL;
 
 #ifdef USE_ANSI_COLOR
     Linecolor *colorBuffer = NULL;
 #endif
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
 
     if (buf->pagerSource == NULL)
 	return NULL;
@@ -8101,10 +8101,10 @@ save2tmp(URLFile uf, char *tmpf)
 {
     FILE *ff;
     clen_t linelen = 0, trbyte = 0;
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
     static JMP_BUF env_bak;
-    volatile int retval = 0;
-    char *volatile buf = NULL;
+    int retval = 0;
+    char *buf = NULL;
 
     ff = fopen(tmpf, "wb");
     if (ff == NULL) {
--- a/frame.c
+++ b/frame.c
@@ -407,7 +407,7 @@ createFrameFile(struct frameset *f, FILE
 #endif
     char *d_target, *p_target, *s_target, *t_target;
     ParsedURL *currentURL, base;
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
     int flag;
 
     if (f == NULL)
--- a/ftp.c
+++ b/ftp.c
@@ -471,11 +471,11 @@ loadFTPDir0(ParsedURL *pu)
     Str FTPDIRtmp;
     Str tmp;
     int status;
-    volatile int sv_type;
+    int sv_type;
     char *realpathname, *fn, *q;
     char **flist;
     int i, nfile, nfile_max;
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
 #ifdef USE_M17N
     wc_ces doc_charset = DocumentCharset;
 
--- a/main.c
+++ b/main.c
@@ -1376,7 +1376,7 @@ cmp_anchor_hseq(const void *a, const voi
 static void
 do_dump(Buffer *buf)
 {
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
 
     prevtrap = mySignal(SIGINT, intTrap);
     if (SETJMP(IntReturn) != 0) {
@@ -1768,10 +1768,10 @@ clear_mark(Line *l)
 
 /* search by regular expression */
 static int
-srchcore(char *volatile str, int (*func) (Buffer *, char *))
+srchcore(char *str, int (*func) (Buffer *, char *))
 {
     MySignalHandler(*prevtrap) ();
-    volatile int i, result = SR_NOTFOUND;
+    int i, result = SR_NOTFOUND;
 
     if (str != NULL && str != SearchString)
 	SearchString = str;
--- a/news.c
+++ b/news.c
@@ -317,15 +317,15 @@ Str
 loadNewsgroup0(ParsedURL *pu)
 #endif
 {
-    volatile Str page;
+    Str page;
     Str tmp;
     URLFile f;
     Buffer *buf;
     char *qgroup, *p, *q, *s, *t, *n;
-    char *volatile scheme, *volatile group, *volatile list;
+    char *scheme, *group, *list;
     int status, i, first, last;
-    volatile int flag = 0, start = 0, end = 0;
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    int flag = 0, start = 0, end = 0;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
 #ifdef USE_M17N
     wc_ces doc_charset = DocumentCharset, mime_charset;
 
--- a/terms.c
+++ b/terms.c
@@ -552,9 +552,9 @@ put_image_kitty(char *url, int x, int y,
     int c, i, j, m, t, is_anim;
     struct stat st;
     pid_t pid;
-    MySignalHandler(*volatile previntr) (SIGNAL_ARG);
-    MySignalHandler(*volatile prevquit) (SIGNAL_ARG);
-    MySignalHandler(*volatile prevstop) (SIGNAL_ARG);
+    MySignalHandler(*previntr) (SIGNAL_ARG);
+    MySignalHandler(*prevquit) (SIGNAL_ARG);
+    MySignalHandler(*prevstop) (SIGNAL_ARG);
 
     if (!url)
 	return;
@@ -770,9 +770,9 @@ put_image_sixel(char *url, int x, int y,
 {
     pid_t pid;
     int do_anim;
-    MySignalHandler(*volatile previntr) (SIGNAL_ARG);
-    MySignalHandler(*volatile prevquit) (SIGNAL_ARG);
-    MySignalHandler(*volatile prevstop) (SIGNAL_ARG);
+    MySignalHandler(*previntr) (SIGNAL_ARG);
+    MySignalHandler(*prevquit) (SIGNAL_ARG);
+    MySignalHandler(*prevstop) (SIGNAL_ARG);
 
     MOVE(y,x);
     flush_tty();
--- a/url.c
+++ b/url.c
@@ -539,7 +539,7 @@ int
 openSocket(char *const hostname,
 	   char *remoteport_name, unsigned short remoteport_num)
 {
-    volatile int sock = -1;
+    int sock = -1;
 #ifdef INET6
     int *af;
     struct addrinfo hints, *res0, *res;
@@ -553,7 +553,7 @@ openSocket(char *const hostname,
     int a1, a2, a3, a4;
     unsigned long adr;
 #endif				/* not INET6 */
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
 
     if (fmInitialized) {
 	/* FIXME: gettextize? */
@@ -2125,8 +2125,8 @@ int
 check_no_proxy(char *domain)
 {
     TextListItem *tl;
-    volatile int ret = 0;
-    MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
+    int ret = 0;
+    MySignalHandler(*prevtrap) (SIGNAL_ARG) = NULL;
 
     if (NO_proxy_domains == NULL || NO_proxy_domains->nitem == 0 ||
 	domain == NULL)
