---
 file.c  |   56 +++++++++++++++++++++++---------------------------------
 menu.c  |   12 ++++++------
 proto.h |   11 +++++------
 table.c |   16 ++++++++--------
 4 files changed, 42 insertions(+), 53 deletions(-)

--- a/file.c
+++ b/file.c
@@ -1961,7 +1961,7 @@ loadGeneralFile(char *path, ParsedURL *v
 		page = loadGopherDir(&f, &pu, &charset);
 		t = "gopher:directory";
 	    } else {
-		page = loadGopherSearch(&f, &pu, &charset);
+		page = loadGopherSearch(&pu, &charset);
 		t = "gopher:search";
 	    }
 	    TRAP_OFF;
@@ -2994,7 +2994,7 @@ flushline(struct html_feed_environ *h_en
 
 void
 do_blankline(struct html_feed_environ *h_env, struct readbuffer *obuf,
-	     int indent, int indent_incr, int width)
+	     int indent, int width)
 {
     if (h_env->blank_lines == 0)
 	flushline(h_env, obuf, indent, 1, width);
@@ -3145,7 +3145,7 @@ process_title(struct parsed_tag *tag)
 }
 
 static Str
-process_n_title(struct parsed_tag *tag)
+process_n_title(void)
 {
     Str tmp;
 
@@ -4494,8 +4494,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	CLOSE_A;
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 1, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	}
 	obuf->flag |= RB_IGNORE_P;
 	if (cmd == HTML_P) {
@@ -4512,8 +4511,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
     case HTML_H:
 	if (!(obuf->flag & (RB_PREMODE | RB_IGNORE_P))) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	}
 	HTMLlineproc1("<b>", h_env);
 	set_alignment(obuf, tag);
@@ -4523,7 +4521,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	if (!(obuf->flag & RB_PREMODE)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
 	}
-	do_blankline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
+	do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	RB_RESTORE_FLAG(obuf);
 	close_anchor(h_env, obuf);
 	obuf->flag |= RB_IGNORE_P;
@@ -4536,8 +4534,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
 	    if (!(obuf->flag & RB_PREMODE) &&
 		(h_env->envc == 0 || cmd == HTML_BLQ))
-		do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			     h_env->limit);
+		do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	}
 	PUSH_ENV(cmd);
 	if (cmd == HTML_UL || cmd == HTML_OL) {
@@ -4569,8 +4566,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	    if (!(obuf->flag & RB_PREMODE) &&
 		(h_env->envc == 0 || cmd == HTML_N_BLQ)) {
 		do_blankline(h_env, obuf,
-			     envs[h_env->envc].indent,
-			     INDENT_INCR, h_env->limit);
+			     envs[h_env->envc].indent, h_env->limit);
 		obuf->flag |= RB_IGNORE_P;
 	    }
 	}
@@ -4583,7 +4579,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	    if (!(obuf->flag & RB_PREMODE) && envs[h_env->envc].env != HTML_DL
 		    && envs[h_env->envc].env != HTML_DL_COMPACT
 		    && envs[h_env->envc].env != HTML_DD)
-		do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
+		do_blankline(h_env, obuf, envs[h_env->envc].indent,
 			     h_env->limit);
 	}
 	PUSH_ENV_NOINDENT(cmd);
@@ -4734,7 +4730,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	    return 1;
 	obuf->flag &= ~RB_TITLE;
 	obuf->end_tag = 0;
-	tmp = process_n_title(tag);
+	tmp = process_n_title();
 	if (tmp)
 	    HTMLlineproc1(tmp->ptr, h_env);
 	return 1;
@@ -4791,7 +4787,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
 	    if (!x)
-		do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
+		do_blankline(h_env, obuf, envs[h_env->envc].indent,
 			     h_env->limit);
 	}
 	else
@@ -4802,8 +4798,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
     case HTML_N_PRE:
 	flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
 	if (!(obuf->flag & RB_IGNORE_P)) {
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	    obuf->flag |= RB_IGNORE_P;
 	    h_env->blank_lines++;
 	}
@@ -4840,8 +4835,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	CLOSE_A;
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	}
 	obuf->flag |= (RB_PRE | RB_IGNORE_P);
 	return 1;
@@ -4849,8 +4843,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	CLOSE_A;
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	    obuf->flag |= RB_IGNORE_P;
 	}
 	obuf->flag &= ~RB_PRE;
@@ -4861,8 +4854,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	CLOSE_A;
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	}
 	obuf->flag |= (RB_PLAIN | RB_IGNORE_P);
 	switch (cmd) {
@@ -4882,8 +4874,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	CLOSE_A;
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	    obuf->flag |= RB_IGNORE_P;
 	}
 	obuf->flag &= ~RB_PLAIN;
@@ -5237,7 +5228,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 		tmp = Sprintf("Refresh (%d sec)", refresh_interval);
 	    if (tmp) {
 		HTMLlineproc1(tmp->ptr, h_env);
-		do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
+		do_blankline(h_env, obuf, envs[h_env->envc].indent,
 			     h_env->limit);
 		if (!is_redisplay &&
 		    !((obuf->flag & RB_NOFRAMES) && RenderFrame)) {
@@ -6288,8 +6279,7 @@ proc_escape(struct readbuffer *obuf, cha
 
 
 static int
-need_flushline(struct html_feed_environ *h_env, struct readbuffer *obuf,
-	       Lineprop mode)
+need_flushline(struct html_feed_environ *h_env, struct readbuffer *obuf)
 {
     char ch;
 
@@ -6488,7 +6478,7 @@ HTMLlineproc0(char *line, struct html_fe
 		if (tbl->vspace > 0 && !(obuf->flag & RB_IGNORE_P)) {
 		    int indent = h_env->envs[h_env->envc].indent;
 		    flushline(h_env, obuf, indent, 0, h_env->limit);
-		    do_blankline(h_env, obuf, indent, 0, h_env->limit);
+		    do_blankline(h_env, obuf, indent,h_env->limit);
 		}
 		save_fonteffect(h_env, obuf);
 		initRenderTable();
@@ -6497,7 +6487,7 @@ HTMLlineproc0(char *line, struct html_fe
 		obuf->flag &= ~RB_IGNORE_P;
 		if (tbl->vspace > 0) {
 		    int indent = h_env->envs[h_env->envc].indent;
-		    do_blankline(h_env, obuf, indent, 0, h_env->limit);
+		    do_blankline(h_env, obuf, indent, h_env->limit);
 		    obuf->flag |= RB_IGNORE_P;
 		}
 		set_space_to_prevchar(obuf->prevchar);
@@ -6643,7 +6633,7 @@ HTMLlineproc0(char *line, struct html_fe
 				   mode);
 		}
 	    }
-	    if (need_flushline(h_env, obuf, mode)) {
+	    if (need_flushline(h_env, obuf)) {
 		char *bp = obuf->line->ptr + obuf->bp.len;
 		char *tp = bp - obuf->bp.tlen;
 		int i = 0;
@@ -7464,10 +7454,10 @@ loadGopherDir0(URLFile *uf, ParsedURL *p
 
 #ifdef USE_M17N
 Str
-loadGopherSearch(URLFile *uf, ParsedURL *pu, wc_ces * charset)
+loadGopherSearch(ParsedURL *pu, wc_ces * charset)
 #else
 Str
-loadGopherSearch0(URLFile *uf, ParsedURL *pu)
+loadGopherSearch0(ParsedURL *pu)
 #endif
 {
     Str tmp;
--- a/menu.c
+++ b/menu.c
@@ -348,7 +348,7 @@ new_menu(Menu *menu, MenuItem *item)
 }
 
 void
-geom_menu(Menu *menu, int x, int y, int mselect)
+geom_menu(Menu *menu, int mselect)
 {
     int win_x, win_y, win_w, win_h;
 
@@ -633,7 +633,7 @@ popup_menu(Menu *parent, Menu *menu)
 	menu->cursorY = parent->cursorY;
 	guess_menu_xy(parent, menu->width, &menu->x, &menu->y);
     }
-    geom_menu(menu, menu->x, menu->y, menu->select);
+    geom_menu(menu, menu->select);
 
     CurrentMenu = menu;
     while (active) {
@@ -1492,7 +1492,7 @@ smDelBuf(char c)
     CurrentMenu->x = x;
     CurrentMenu->y = y;
 
-    geom_menu(CurrentMenu, x, y, 0);
+    geom_menu(CurrentMenu, 0);
 
     CurrentMenu->select = (mselect <= CurrentMenu->nitem - 2) ? mselect
 	: (CurrentMenu->nitem - 2);
@@ -1528,7 +1528,7 @@ smUpBuf(char c)
     initSelectMenu();
     CurrentMenu->x = x;
     CurrentMenu->y = y;
-    geom_menu(CurrentMenu, 0, 0, 0);
+    geom_menu(CurrentMenu, 0);
     CurrentMenu->select = --mselect;
     displayBuffer(Currentbuf, B_FORCE_REDRAW);
     draw_all_menu(CurrentMenu);
@@ -1566,7 +1566,7 @@ smDnBuf(char c)
     initSelectMenu();
     CurrentMenu->x = x;
     CurrentMenu->y = y;
-    geom_menu(CurrentMenu, 0, 0, 0);
+    geom_menu(CurrentMenu, 0);
     CurrentMenu->select = ++mselect;
     displayBuffer(Currentbuf, B_FORCE_REDRAW);
     draw_all_menu(CurrentMenu);
@@ -1706,7 +1706,7 @@ smDelTab(char c)
     CurrentMenu->x = x;
     CurrentMenu->y = y;
 
-    geom_menu(CurrentMenu, x, y, 0);
+    geom_menu(CurrentMenu, 0);
 
     CurrentMenu->select = (mselect <= CurrentMenu->nitem - 2) ? mselect
 	: (CurrentMenu->nitem - 2);
--- a/proto.h
+++ b/proto.h
@@ -208,8 +208,7 @@ extern void push_render_image(Str str, i
 extern void flushline(struct html_feed_environ *h_env, struct readbuffer *obuf,
 		      int indent, int force, int width);
 extern void do_blankline(struct html_feed_environ *h_env,
-			 struct readbuffer *obuf, int indent, int indent_incr,
-			 int width);
+			 struct readbuffer *obuf, int indent, int width);
 extern void purgeline(struct html_feed_environ *h_env);
 extern void save_fonteffect(struct html_feed_environ *h_env,
 			    struct readbuffer *obuf);
@@ -257,12 +256,12 @@ extern Buffer *loadHTMLString(Str page);
 #ifdef USE_GOPHER
 #ifdef USE_M17N
 extern Str loadGopherDir(URLFile *uf, ParsedURL *pu, wc_ces * charset);
-extern Str loadGopherSearch(URLFile *uf, ParsedURL *pu, wc_ces * charset);
+extern Str loadGopherSearch(ParsedURL *pu, wc_ces * charset);
 #else
 extern Str loadGopherDir0(URLFile *uf, ParsedURL *pu);
-extern Str loadGopherSearch0(URLFile *uf, ParsedURL *pu);
+extern Str loadGopherSearch0(ParsedURL *pu);
 #define loadGopherDir(uf,pu,charset) loadGopherDir0(uf,pu)
-#define loadGopherSearch(uf,pu,charset) loadGopherSearch0(uf,pu)
+#define loadGopherSearch(pu,charset) loadGopherSearch0(pu)
 #endif
 #endif				/* USE_GOPHER */
 extern Buffer *loadBuffer(URLFile *uf, Buffer *newBuf);
@@ -735,7 +734,7 @@ extern void initMouseAction(void);
 
 #ifdef USE_MENU
 extern void new_menu(Menu *menu, MenuItem *item);
-extern void geom_menu(Menu *menu, int x, int y, int mselect);
+extern void geom_menu(Menu *menu, int mselect);
 extern void draw_all_menu(Menu *menu);
 extern void draw_menu(Menu *menu);
 extern void draw_menu_item(Menu *menu, int mselect);
--- a/table.c
+++ b/table.c
@@ -754,7 +754,7 @@ do_refill(struct table *tbl, int row, in
     if (h_env.limit > maxlimit)
 	h_env.limit = maxlimit;
     if (tbl->border_mode != BORDER_NONE && tbl->vcellpadding > 0)
-	do_blankline(&h_env, &obuf, 0, 0, h_env.limit);
+	do_blankline(&h_env, &obuf, 0, h_env.limit);
     for (l = orgdata->first; l != NULL; l = l->next) {
 	if (TAG_IS(l->ptr, "<table_alt", 10)) {
 	    int id = -1;
@@ -771,7 +771,7 @@ do_refill(struct table *tbl, int row, in
 		save_fonteffect(&h_env, h_env.obuf);
 		flushline(&h_env, &obuf, 0, 2, h_env.limit);
 		if (t->vspace > 0 && !(obuf.flag & RB_IGNORE_P))
-		    do_blankline(&h_env, &obuf, 0, 0, h_env.limit);
+		    do_blankline(&h_env, &obuf, 0, h_env.limit);
 		if (RB_GET_ALIGN(h_env.obuf) == RB_CENTER)
 		    alignment = ALIGN_CENTER;
 		else if (RB_GET_ALIGN(h_env.obuf) == RB_RIGHT)
@@ -791,7 +791,7 @@ do_refill(struct table *tbl, int row, in
 		obuf.flag &= ~RB_IGNORE_P;
 		h_env.blank_lines = 0;
 		if (t->vspace > 0) {
-		    do_blankline(&h_env, &obuf, 0, 0, h_env.limit);
+		    do_blankline(&h_env, &obuf, 0, h_env.limit);
 		    obuf.flag |= RB_IGNORE_P;
 		}
 	    }
@@ -809,7 +809,7 @@ do_refill(struct table *tbl, int row, in
 	int rowspan = table_rowspan(tbl, row, col);
 	if (row + rowspan <= tbl->maxrow) {
 	    if (tbl->vcellpadding > 0 && !(obuf.flag & RB_IGNORE_P))
-		do_blankline(&h_env, &obuf, 0, 0, h_env.limit);
+		do_blankline(&h_env, &obuf, 0, h_env.limit);
 	}
 	else {
 	    if (tbl->vspace > 0)
@@ -819,7 +819,7 @@ do_refill(struct table *tbl, int row, in
     else {
 	if (tbl->vcellpadding > 0) {
 	    if (!(obuf.flag & RB_IGNORE_P))
-		do_blankline(&h_env, &obuf, 0, 0, h_env.limit);
+		do_blankline(&h_env, &obuf, 0, h_env.limit);
 	}
 	else
 	    purgeline(&h_env);
@@ -2156,7 +2156,7 @@ check_minimum0(struct table *t, int min)
 }
 
 static int
-setwidth0(struct table *t, struct table_mode *mode)
+setwidth0(struct table *t)
 {
     int w;
     int width = t->tabcontentssize;
@@ -2184,7 +2184,7 @@ setwidth0(struct table *t, struct table_
 static void
 setwidth(struct table *t, struct table_mode *mode)
 {
-    int width = setwidth0(t, mode);
+    int width = setwidth0(t);
     if (width < 0)
 	return;
 #ifdef NOWRAP
@@ -3121,7 +3121,7 @@ feed_table_tag(struct table *tbl, char *
 		    tbl->fixed_width[tbl->col] = w;
 	    }
 #endif
-	    setwidth0(tbl, mode);
+	    setwidth0(tbl);
 	    clearcontentssize(tbl, mode);
 	}
 	break;
