From b7d96beed4598b1d857eb75f300267a720024737 Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Thu, 2 Sep 2021 20:37:35 +0200
Subject: [PATCH 071/172] Remove unused typedef's

---
 istream.h |    6 ------
 1 file changed, 6 deletions(-)

--- a/istream.h
+++ b/istream.h
@@ -100,12 +100,6 @@ union input_stream {
 };
 
 typedef struct base_stream *BaseStream;
-typedef struct file_stream *FileStream;
-typedef struct str_stream *StrStream;
-#ifdef USE_SSL
-typedef struct ssl_stream *SSLStream;
-#endif				/* USE_SSL */
-typedef struct encoded_stream *EncodedStrStream;
 
 typedef union input_stream *InputStream;
 
