---
 file.c  |   56 +++++++++++++++++++++++---------------------------------
 proto.h |    9 ++++-----
 table.c |   16 ++++++++--------
 3 files changed, 35 insertions(+), 46 deletions(-)

--- a/file.c
+++ b/file.c
@@ -1961,7 +1961,7 @@ loadGeneralFile(char *path, ParsedURL *v
 		page = loadGopherDir(&f, &pu, &charset);
 		t = "gopher:directory";
 	    } else {
-		page = loadGopherSearch(&f, &pu, &charset);
+		page = loadGopherSearch(&pu, &charset);
 		t = "gopher:search";
 	    }
 	    TRAP_OFF;
@@ -2985,7 +2985,7 @@ flushline(struct html_feed_environ *h_en
 
 void
 do_blankline(struct html_feed_environ *h_env, struct readbuffer *obuf,
-	     int indent, int indent_incr, int width)
+	     int indent, int width)
 {
     if (h_env->blank_lines == 0)
 	flushline(h_env, obuf, indent, 1, width);
@@ -3136,7 +3136,7 @@ process_title(struct parsed_tag *tag)
 }
 
 static Str
-process_n_title(struct parsed_tag *tag)
+process_n_title(void)
 {
     Str tmp;
 
@@ -4485,8 +4485,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	CLOSE_A;
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 1, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	}
 	obuf->flag |= RB_IGNORE_P;
 	if (cmd == HTML_P) {
@@ -4503,8 +4502,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
     case HTML_H:
 	if (!(obuf->flag & (RB_PREMODE | RB_IGNORE_P))) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	}
 	HTMLlineproc1("<b>", h_env);
 	set_alignment(obuf, tag);
@@ -4514,7 +4512,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	if (!(obuf->flag & RB_PREMODE)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
 	}
-	do_blankline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
+	do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	RB_RESTORE_FLAG(obuf);
 	close_anchor(h_env, obuf);
 	obuf->flag |= RB_IGNORE_P;
@@ -4527,8 +4525,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
 	    if (!(obuf->flag & RB_PREMODE) &&
 		(h_env->envc == 0 || cmd == HTML_BLQ))
-		do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			     h_env->limit);
+		do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	}
 	PUSH_ENV(cmd);
 	if (cmd == HTML_UL || cmd == HTML_OL) {
@@ -4560,8 +4557,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	    if (!(obuf->flag & RB_PREMODE) &&
 		(h_env->envc == 0 || cmd == HTML_N_BLQ)) {
 		do_blankline(h_env, obuf,
-			     envs[h_env->envc].indent,
-			     INDENT_INCR, h_env->limit);
+			     envs[h_env->envc].indent, h_env->limit);
 		obuf->flag |= RB_IGNORE_P;
 	    }
 	}
@@ -4574,7 +4570,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	    if (!(obuf->flag & RB_PREMODE) && envs[h_env->envc].env != HTML_DL
 		    && envs[h_env->envc].env != HTML_DL_COMPACT
 		    && envs[h_env->envc].env != HTML_DD)
-		do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
+		do_blankline(h_env, obuf, envs[h_env->envc].indent,
 			     h_env->limit);
 	}
 	PUSH_ENV_NOINDENT(cmd);
@@ -4725,7 +4721,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	    return 1;
 	obuf->flag &= ~RB_TITLE;
 	obuf->end_tag = 0;
-	tmp = process_n_title(tag);
+	tmp = process_n_title();
 	if (tmp)
 	    HTMLlineproc1(tmp->ptr, h_env);
 	return 1;
@@ -4782,7 +4778,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
 	    if (!x)
-		do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
+		do_blankline(h_env, obuf, envs[h_env->envc].indent,
 			     h_env->limit);
 	}
 	else
@@ -4793,8 +4789,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
     case HTML_N_PRE:
 	flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
 	if (!(obuf->flag & RB_IGNORE_P)) {
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	    obuf->flag |= RB_IGNORE_P;
 	    h_env->blank_lines++;
 	}
@@ -4831,8 +4826,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	CLOSE_A;
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	}
 	obuf->flag |= (RB_PRE | RB_IGNORE_P);
 	return 1;
@@ -4840,8 +4834,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	CLOSE_A;
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	    obuf->flag |= RB_IGNORE_P;
 	}
 	obuf->flag &= ~RB_PRE;
@@ -4852,8 +4845,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	CLOSE_A;
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	}
 	obuf->flag |= (RB_PLAIN | RB_IGNORE_P);
 	switch (cmd) {
@@ -4873,8 +4865,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 	CLOSE_A;
 	if (!(obuf->flag & RB_IGNORE_P)) {
 	    flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
-	    do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
-			 h_env->limit);
+	    do_blankline(h_env, obuf, envs[h_env->envc].indent, h_env->limit);
 	    obuf->flag |= RB_IGNORE_P;
 	}
 	obuf->flag &= ~RB_PLAIN;
@@ -5228,7 +5219,7 @@ HTMLtagproc1(struct parsed_tag *tag, str
 		tmp = Sprintf("Refresh (%d sec)", refresh_interval);
 	    if (tmp) {
 		HTMLlineproc1(tmp->ptr, h_env);
-		do_blankline(h_env, obuf, envs[h_env->envc].indent, 0,
+		do_blankline(h_env, obuf, envs[h_env->envc].indent,
 			     h_env->limit);
 		if (!is_redisplay &&
 		    !((obuf->flag & RB_NOFRAMES) && RenderFrame)) {
@@ -6279,8 +6270,7 @@ proc_escape(struct readbuffer *obuf, cha
 
 
 static int
-need_flushline(struct html_feed_environ *h_env, struct readbuffer *obuf,
-	       Lineprop mode)
+need_flushline(struct html_feed_environ *h_env, struct readbuffer *obuf)
 {
     char ch;
 
@@ -6479,7 +6469,7 @@ HTMLlineproc0(char *line, struct html_fe
 		if (tbl->vspace > 0 && !(obuf->flag & RB_IGNORE_P)) {
 		    int indent = h_env->envs[h_env->envc].indent;
 		    flushline(h_env, obuf, indent, 0, h_env->limit);
-		    do_blankline(h_env, obuf, indent, 0, h_env->limit);
+		    do_blankline(h_env, obuf, indent,h_env->limit);
 		}
 		save_fonteffect(h_env, obuf);
 		initRenderTable();
@@ -6488,7 +6478,7 @@ HTMLlineproc0(char *line, struct html_fe
 		obuf->flag &= ~RB_IGNORE_P;
 		if (tbl->vspace > 0) {
 		    int indent = h_env->envs[h_env->envc].indent;
-		    do_blankline(h_env, obuf, indent, 0, h_env->limit);
+		    do_blankline(h_env, obuf, indent, h_env->limit);
 		    obuf->flag |= RB_IGNORE_P;
 		}
 		set_space_to_prevchar(obuf->prevchar);
@@ -6634,7 +6624,7 @@ HTMLlineproc0(char *line, struct html_fe
 				   mode);
 		}
 	    }
-	    if (need_flushline(h_env, obuf, mode)) {
+	    if (need_flushline(h_env, obuf)) {
 		char *bp = obuf->line->ptr + obuf->bp.len;
 		char *tp = bp - obuf->bp.tlen;
 		int i = 0;
@@ -7455,10 +7445,10 @@ loadGopherDir0(URLFile *uf, ParsedURL *p
 
 #ifdef USE_M17N
 Str
-loadGopherSearch(URLFile *uf, ParsedURL *pu, wc_ces * charset)
+loadGopherSearch(ParsedURL *pu, wc_ces * charset)
 #else
 Str
-loadGopherSearch0(URLFile *uf, ParsedURL *pu)
+loadGopherSearch0(ParsedURL *pu)
 #endif
 {
     Str tmp;
--- a/proto.h
+++ b/proto.h
@@ -208,8 +208,7 @@ extern void push_render_image(Str str, i
 extern void flushline(struct html_feed_environ *h_env, struct readbuffer *obuf,
 		      int indent, int force, int width);
 extern void do_blankline(struct html_feed_environ *h_env,
-			 struct readbuffer *obuf, int indent, int indent_incr,
-			 int width);
+			 struct readbuffer *obuf, int indent, int width);
 extern void purgeline(struct html_feed_environ *h_env);
 extern void save_fonteffect(struct html_feed_environ *h_env,
 			    struct readbuffer *obuf);
@@ -257,12 +256,12 @@ extern Buffer *loadHTMLString(Str page);
 #ifdef USE_GOPHER
 #ifdef USE_M17N
 extern Str loadGopherDir(URLFile *uf, ParsedURL *pu, wc_ces * charset);
-extern Str loadGopherSearch(URLFile *uf, ParsedURL *pu, wc_ces * charset);
+extern Str loadGopherSearch(ParsedURL *pu, wc_ces * charset);
 #else
 extern Str loadGopherDir0(URLFile *uf, ParsedURL *pu);
-extern Str loadGopherSearch0(URLFile *uf, ParsedURL *pu);
+extern Str loadGopherSearch0(ParsedURL *pu);
 #define loadGopherDir(uf,pu,charset) loadGopherDir0(uf,pu)
-#define loadGopherSearch(uf,pu,charset) loadGopherSearch0(uf,pu)
+#define loadGopherSearch(pu,charset) loadGopherSearch0(pu)
 #endif
 #endif				/* USE_GOPHER */
 extern Buffer *loadBuffer(URLFile *uf, Buffer *newBuf);
--- a/table.c
+++ b/table.c
@@ -754,7 +754,7 @@ do_refill(struct table *tbl, int row, in
     if (h_env.limit > maxlimit)
 	h_env.limit = maxlimit;
     if (tbl->border_mode != BORDER_NONE && tbl->vcellpadding > 0)
-	do_blankline(&h_env, &obuf, 0, 0, h_env.limit);
+	do_blankline(&h_env, &obuf, 0, h_env.limit);
     for (l = orgdata->first; l != NULL; l = l->next) {
 	if (TAG_IS(l->ptr, "<table_alt", 10)) {
 	    int id = -1;
@@ -771,7 +771,7 @@ do_refill(struct table *tbl, int row, in
 		save_fonteffect(&h_env, h_env.obuf);
 		flushline(&h_env, &obuf, 0, 2, h_env.limit);
 		if (t->vspace > 0 && !(obuf.flag & RB_IGNORE_P))
-		    do_blankline(&h_env, &obuf, 0, 0, h_env.limit);
+		    do_blankline(&h_env, &obuf, 0, h_env.limit);
 		if (RB_GET_ALIGN(h_env.obuf) == RB_CENTER)
 		    alignment = ALIGN_CENTER;
 		else if (RB_GET_ALIGN(h_env.obuf) == RB_RIGHT)
@@ -791,7 +791,7 @@ do_refill(struct table *tbl, int row, in
 		obuf.flag &= ~RB_IGNORE_P;
 		h_env.blank_lines = 0;
 		if (t->vspace > 0) {
-		    do_blankline(&h_env, &obuf, 0, 0, h_env.limit);
+		    do_blankline(&h_env, &obuf, 0, h_env.limit);
 		    obuf.flag |= RB_IGNORE_P;
 		}
 	    }
@@ -809,7 +809,7 @@ do_refill(struct table *tbl, int row, in
 	int rowspan = table_rowspan(tbl, row, col);
 	if (row + rowspan <= tbl->maxrow) {
 	    if (tbl->vcellpadding > 0 && !(obuf.flag & RB_IGNORE_P))
-		do_blankline(&h_env, &obuf, 0, 0, h_env.limit);
+		do_blankline(&h_env, &obuf, 0, h_env.limit);
 	}
 	else {
 	    if (tbl->vspace > 0)
@@ -819,7 +819,7 @@ do_refill(struct table *tbl, int row, in
     else {
 	if (tbl->vcellpadding > 0) {
 	    if (!(obuf.flag & RB_IGNORE_P))
-		do_blankline(&h_env, &obuf, 0, 0, h_env.limit);
+		do_blankline(&h_env, &obuf, 0, h_env.limit);
 	}
 	else
 	    purgeline(&h_env);
@@ -2156,7 +2156,7 @@ check_minimum0(struct table *t, int min)
 }
 
 static int
-setwidth0(struct table *t, struct table_mode *mode)
+setwidth0(struct table *t)
 {
     int w;
     int width = t->tabcontentssize;
@@ -2184,7 +2184,7 @@ setwidth0(struct table *t, struct table_
 static void
 setwidth(struct table *t, struct table_mode *mode)
 {
-    int width = setwidth0(t, mode);
+    int width = setwidth0(t);
     if (width < 0)
 	return;
 #ifdef NOWRAP
@@ -3121,7 +3121,7 @@ feed_table_tag(struct table *tbl, char *
 		    tbl->fixed_width[tbl->col] = w;
 	    }
 #endif
-	    setwidth0(tbl, mode);
+	    setwidth0(tbl);
 	    clearcontentssize(tbl, mode);
 	}
 	break;
