From: Rene Kita <mail@rkta.de>
Date: Sat Oct 28 13:02:35 CEST 2023
Subject: Replace AC_TRY_CPP
---
 configure    |   84 ++++++++++++++++++++++++++++-------------------------------
 configure.ac |    4 +-
 2 files changed, 43 insertions(+), 45 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -98,9 +98,9 @@ if test x"$enable_mouse" = xyes; then
  AC_CHECK_LIB(gpm, Gpm_Open, [AC_DEFINE(USE_GPM)
    		W3M_LIBS="$W3M_LIBS -lgpm"])
  AC_SUBST(USE_SYSMOUSE)
- AC_TRY_CPP([#include <sys/consio.h>
+ AC_CHECK_HEADER([#include <sys/consio.h>
    MOUSE_GETINFO], [AC_DEFINE(USE_SYSMOUSE)],
-   [AC_TRY_CPP([#include <machine/console.h>
+   [AC_CHECK_HEADER([#include <machine/console.h>
       MOUSE_GETINFO], [AC_DEFINE(USE_SYSMOUSE)])])
 fi
 AC_SUBST(AUXBIN_TARGETS)
--- a/configure
+++ b/configure
@@ -9733,29 +9733,56 @@ then :
 fi
 
 
- cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <sys/consio.h>
-   MOUSE_GETINFO
-_ACEOF
-if ac_fn_c_try_cpp "$LINENO"
+ ac_header= ac_cache=
+for ac_item in $ac_header_c_list
+do
+  if test $ac_cache; then
+    ac_fn_c_check_header_compile "$LINENO" $ac_header ac_cv_header_$ac_cache "$ac_includes_default"
+    if eval test \"x\$ac_cv_header_$ac_cache\" = xyes; then
+      printf "%s\n" "#define $ac_item 1" >> confdefs.h
+    fi
+    ac_header= ac_cache=
+  elif test $ac_header; then
+    ac_cache=$ac_item
+  else
+    ac_header=$ac_item
+  fi
+done
+
+
+
+
+
+
+
+
+if test $ac_cv_header_stdlib_h = yes && test $ac_cv_header_string_h = yes
+then :
+
+printf "%s\n" "#define STDC_HEADERS 1" >>confdefs.h
+
+fi
+as_ac_Header=`printf "%s\n" "ac_cv_header_#include <sys/consio.h>
+   MOUSE_GETINFO" | $as_tr_sh`
+ac_fn_c_check_header_compile "$LINENO" "#include <sys/consio.h>
+   MOUSE_GETINFO" "$as_ac_Header" "$ac_includes_default"
+if eval test \"x\$"$as_ac_Header"\" = x"yes"
 then :
   printf "%s\n" "#define USE_SYSMOUSE 1" >>confdefs.h
 
 else $as_nop
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <machine/console.h>
-      MOUSE_GETINFO
-_ACEOF
-if ac_fn_c_try_cpp "$LINENO"
+  as_ac_Header=`printf "%s\n" "ac_cv_header_#include <machine/console.h>
+      MOUSE_GETINFO" | $as_tr_sh`
+ac_fn_c_check_header_compile "$LINENO" "#include <machine/console.h>
+      MOUSE_GETINFO" "$as_ac_Header" "$ac_includes_default"
+if eval test \"x\$"$as_ac_Header"\" = x"yes"
 then :
   printf "%s\n" "#define USE_SYSMOUSE 1" >>confdefs.h
 
 fi
-rm -f conftest.err conftest.i conftest.$ac_ext
+
 fi
-rm -f conftest.err conftest.i conftest.$ac_ext
+
 fi
 
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking terminal library" >&5
@@ -9817,35 +9844,6 @@ fi
 
  done
 
-ac_header= ac_cache=
-for ac_item in $ac_header_c_list
-do
-  if test $ac_cache; then
-    ac_fn_c_check_header_compile "$LINENO" $ac_header ac_cv_header_$ac_cache "$ac_includes_default"
-    if eval test \"x\$ac_cv_header_$ac_cache\" = xyes; then
-      printf "%s\n" "#define $ac_item 1" >> confdefs.h
-    fi
-    ac_header= ac_cache=
-  elif test $ac_header; then
-    ac_cache=$ac_item
-  else
-    ac_header=$ac_item
-  fi
-done
-
-
-
-
-
-
-
-
-if test $ac_cv_header_stdlib_h = yes && test $ac_cv_header_string_h = yes
-then :
-
-printf "%s\n" "#define STDC_HEADERS 1" >>confdefs.h
-
-fi
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking GC library exists" >&5
 printf %s "checking GC library exists... " >&6; }
 
