---
 scripts/w3mhelp.cgi.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/scripts/w3mhelp.cgi.in
+++ b/scripts/w3mhelp.cgi.in
@@ -269,7 +269,7 @@ PAGE
     foreach $fid (@list) {
 	local ($name, $desc, $keys) = split("\n", $funckeydesc{$fid}, 3);
 	$keys =~ s/,$//;
-	$keys = '<NOT ASSIGNED>' if ($keys eq '');
+	$keys = '<N/A>' if ($keys eq '');
 	$keys =~ s/[<>&]/$htmlesc{$&}/ge;
 	if ($name =~ /:/) {
 		$name = "";
