---
 display.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/display.c
+++ b/display.c
@@ -5,6 +5,9 @@
 #include "fm.h"
 
 /* *INDENT-OFF* */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wold-style-definition"
+
 #ifdef USE_COLOR
 
 static void effect_anchor_start(void);
@@ -173,6 +176,7 @@ effect_visited_end(void)
 #define EFFECT_MARK_START         standout()
 #define EFFECT_MARK_END           standend()
 #endif				/* not USE_COLOR */
+#pragma GCC diagnostic pop
 /* *INDENT-ON* */
 
 void
