--- a/main.c
+++ b/main.c
@@ -4724,6 +4724,13 @@ DEFUN(vwSrc, SOURCE VIEW, "Toggle betwee
 	displayBuffer(Currentbuf, B_NORMAL);
 	return;
     }
+
+    /* TODO(rkta): keep && document this */
+    if (!strcmp(Currentbuf->type, "text/gemini")) {
+	geminize();
+	return;
+    }
+
     if (Currentbuf->sourcefile == NULL) {
 	if (Currentbuf->pagerSource &&
 	    !strcasecmp(Currentbuf->type, "text/plain")) {
@@ -4796,6 +4803,91 @@ DEFUN(vwSrc, SOURCE VIEW, "Toggle betwee
     buf->clone = Currentbuf->clone;
     (*buf->clone)++;
 
+    buf->need_reshape = TRUE;
+    reshapeBuffer(buf);
+    pushBuffer(buf);
+    displayBuffer(Currentbuf, B_NORMAL);
+}
+
+DEFUN(geminize, GEMINIZE, "Toggle between text/gemini shown or processed")
+{
+    Buffer *buf;
+
+    if (Currentbuf->type == NULL || Currentbuf->bufferprop & BP_FRAME)
+	return;
+    if ((buf = Currentbuf->linkBuffer[LB_SOURCE]) != NULL ||
+	(buf = Currentbuf->linkBuffer[LB_N_SOURCE]) != NULL) {
+	Currentbuf = buf;
+	displayBuffer(Currentbuf, B_NORMAL);
+	return;
+    }
+
+    if (Currentbuf->sourcefile == NULL) {
+	if (Currentbuf->pagerSource &&
+	    !strcasecmp(Currentbuf->type, "text/plain")) {
+#ifdef USE_M17N
+	    wc_ces old_charset;
+	    wc_bool old_fix_width_conv;
+#endif
+	    FILE *f;
+	    Str tmpf = tmpfname(TMPF_SRC, NULL);
+	    f = fopen(tmpf->ptr, "w");
+	    if (f == NULL)
+		return;
+#ifdef USE_M17N
+	    old_charset = DisplayCharset;
+	    old_fix_width_conv = WcOption.fix_width_conv;
+	    DisplayCharset = (Currentbuf->document_charset != WC_CES_US_ASCII)
+		? Currentbuf->document_charset : 0;
+	    WcOption.fix_width_conv = WC_FALSE;
+#endif
+	    saveBufferBody(Currentbuf, f, TRUE);
+#ifdef USE_M17N
+	    DisplayCharset = old_charset;
+	    WcOption.fix_width_conv = old_fix_width_conv;
+#endif
+	    fclose(f);
+	    Currentbuf->sourcefile = tmpf->ptr;
+	}
+	else {
+	    return;
+	}
+    }
+
+    buf = newBuffer(INIT_BUFFER_WIDTH);
+
+    if (!strcmp(Currentbuf->type, "text/gemini")) {
+	buf->type = "text/plain";
+	buf->real_type = Currentbuf->real_type;
+	buf->buffername = Sprintf("source of %s", Currentbuf->buffername)->ptr;
+	buf->linkBuffer[LB_N_SOURCE] = Currentbuf;
+	Currentbuf->linkBuffer[LB_SOURCE] = buf;
+    }
+    else if (!strcasecmp(Currentbuf->type, "text/plain")) {
+	buf->type = "text/gemini";
+	if (Currentbuf->real_type &&
+	    !strcasecmp(Currentbuf->real_type, "text/plain"))
+	    buf->real_type = "text/gemini";
+	else
+	    buf->real_type = Currentbuf->real_type;
+	buf->buffername = Sprintf("Gemini view of %s",
+				  Currentbuf->buffername)->ptr;
+	buf->linkBuffer[LB_SOURCE] = Currentbuf;
+	Currentbuf->linkBuffer[LB_N_SOURCE] = buf;
+    }
+    else {
+	return;
+    }
+    buf->currentURL = Currentbuf->currentURL;
+    buf->real_scheme = Currentbuf->real_scheme;
+    buf->filename = Currentbuf->filename;
+    buf->sourcefile = Currentbuf->sourcefile;
+    buf->header_source = Currentbuf->header_source;
+    buf->search_header = Currentbuf->search_header;
+    buf->document_charset = Currentbuf->document_charset;
+    buf->clone = Currentbuf->clone;
+    (*buf->clone)++;
+
     buf->need_reshape = TRUE;
     reshapeBuffer(buf);
     pushBuffer(buf);
--- a/proto.h
+++ b/proto.h
@@ -96,6 +96,7 @@ extern void peekURL(void);
 extern void peekIMG(void);
 extern void curURL(void);
 extern void vwSrc(void);
+extern void geminize(void);
 extern void reload(void);
 extern void reshape(void);
 extern void chkURL(void);
--- a/buffer.c
+++ b/buffer.c
@@ -488,7 +488,7 @@ reshapeBuffer(Buffer *buf)
     URLFile f;
     Buffer sbuf;
     /* TODO(rkta): Shouldn't we do this for all schemes? */
-    int scheme;
+    Str t;
 #ifdef USE_M17N
     wc_uint8 old_auto_detect = WcOption.auto_detect;
 #endif
@@ -499,7 +499,7 @@ reshapeBuffer(Buffer *buf)
     buf->width = INIT_BUFFER_WIDTH;
     if (buf->sourcefile == NULL)
 	return;
-    scheme = buf->currentURL.scheme;
+    t = Strnew_charp(buf->type);
     init_stream(&f, SCM_LOCAL, NULL);
     examineFile(buf->mailcap_source ? buf->mailcap_source : buf->sourcefile,
 		&f);
@@ -545,7 +545,7 @@ reshapeBuffer(Buffer *buf)
 #endif
     if (is_html_type(buf->type))
 	loadHTMLBuffer(&f, buf);
-    else if (scheme == SCM_GEMINI)
+    else if (!strcmp(t->ptr, "text/gemini"))
 	loadGeminiBuffer(&f, buf);
     else
 	loadBuffer(&f, buf);
