--- a/fm.h
+++ b/fm.h
@@ -444,7 +444,7 @@ typedef struct _LinkList {
 } LinkList;
 
 typedef struct _Buffer {
-    char *filename;
+    const char *filename;
     char *buffername;
     Line *firstLine;
     Line *topLine;
--- a/main.c
+++ b/main.c
@@ -2859,7 +2859,7 @@ cur_real_linenumber(Buffer *buf)
 /* Run editor on the current buffer */
 DEFUN(editBf, EDIT, "Edit local source")
 {
-    char *fn = Currentbuf->filename;
+    const char *fn = Currentbuf->filename;
     Str cmd;
 
     if (fn == NULL || Currentbuf->pagerSource != NULL ||	/* Behaving as a pager */
@@ -6146,7 +6146,7 @@ GetWord(Buffer *buf)
 static void
 execdict(const char *word)
 {
-    char *w, *dictcmd;
+    const char *w, *dictcmd;
     Buffer *buf;
 
     if (!UseDictCommand || word == NULL || *word == '\0') {
--- a/proto.h
+++ b/proto.h
@@ -204,7 +204,7 @@ extern void push_symbol(Str str, char sy
 #ifdef USE_UNICODE
 extern void update_utf8_symbol(void);
 #endif
-extern Buffer *loadGeneralFile(char *path, ParsedURL *current, char *referer,
+extern Buffer *loadGeneralFile(const char *path, ParsedURL *current, char *referer,
 			       int flag, FormList *request, int dl_only);
 extern int is_boundary(const unsigned char *, const unsigned char *);
 extern void push_render_image(Str str, int width, int limit,
@@ -529,7 +529,7 @@ extern struct mailcap *searchMailcap(str
 extern void initMailcap(void);
 extern char *acceptableMimeTypes(void);
 extern struct mailcap *searchExtViewer(const char *type);
-extern Str unquote_mailcap(char *qstr, const char *type, char *name, char *attr,
+extern Str unquote_mailcap(char *qstr, const char *type, const char *name, char *attr,
 			   int *mc_stat);
 extern char *guessContentType(const char *filename);
 extern TextList *make_domain_list(char *domain_list);
--- a/mailcap.c
+++ b/mailcap.c
@@ -349,7 +349,7 @@ quote_mailcap(const char *s, int flag)
 
 
 static Str
-unquote_mailcap_loop(char *qstr, const char *type, char *name, char *attr,
+unquote_mailcap_loop(char *qstr, const char *type, const char *name, char *attr,
 		     int *mc_stat, int flag0)
 {
     Str str, tmp, test, then;
