---
 file.c    |    2 +-
 fm.h      |    1 +
 history.c |    4 +++-
 main.c    |    2 +-
 rc.c      |    3 +++
 5 files changed, 9 insertions(+), 3 deletions(-)

--- a/rc.c
+++ b/rc.c
@@ -78,6 +78,7 @@ static int OptionEncode = FALSE;
 #define CMT_OPEN_TAB_DL_LIST N_("Open download list panel on new tab")
 #define CMT_DISPLINK     N_("Display link URL automatically")
 #define CMT_DISPLINKNUMBER N_("Display link numbers")
+#define CMT_ZEROBASEDLINKNO N_("Use 0 as first link number")
 #define CMT_DECODE_URL   N_("Display decoded URL")
 #define CMT_DISPLINEINFO N_("Display current line information")
 #define CMT_DISP_COLUMN_NUMBER N_("Display column number in line information")
@@ -400,6 +401,8 @@ struct param_ptr params1[] = {
      NULL},
     {"display_link_number", P_INT, PI_ONOFF, (void *)&displayLinkNumber,
      CMT_DISPLINKNUMBER, NULL},
+    {"zero_based_link_no", P_INT, PI_ONOFF, (void *)&zeroBasedLinkNo,
+     CMT_ZEROBASEDLINKNO, NULL},
     {"decode_url", P_INT, PI_ONOFF, (void *)&DecodeURL, CMT_DECODE_URL, NULL},
     {"display_lineinfo", P_INT, PI_ONOFF, (void *)&displayLineInfo,
      CMT_DISPLINEINFO, NULL},
--- a/file.c
+++ b/file.c
@@ -1673,7 +1673,7 @@ checkRedirection(ParsedURL *pu)
 Str
 getLinkNumberStr(int correction)
 {
-    return Sprintf("[%d]", cur_hseq + correction);
+    return Sprintf("[%d]", cur_hseq + correction - !!zeroBasedLinkNo);
 }
 
 /* 
--- a/fm.h
+++ b/fm.h
@@ -972,6 +972,7 @@ global int nextpage_topline init(FALSE);
 global char *displayTitleTerm init(NULL);
 global int displayLink init(FALSE);
 global int displayLinkNumber init(FALSE);
+global int zeroBasedLinkNo init(FALSE);
 global int displayLineInfo init(FALSE);
 global int displayColumnNumber init(FALSE);
 global int DecodeURL init(FALSE);
--- a/main.c
+++ b/main.c
@@ -3619,7 +3619,7 @@ DEFUN(topA, LINK_BEGIN, "Move to the fir
     if (prec_num > hl->nmark)
 	hseq = hl->nmark - 1;
     else if (prec_num > 0)
-	hseq = prec_num - 1;
+	hseq = prec_num - 1 + !!zeroBasedLinkNo;
     do {
 	if (hseq >= hl->nmark)
 	    return;
--- a/history.c
+++ b/history.c
@@ -20,12 +20,14 @@ historyBuffer(Hist *hist)
 {
     Str src = Strnew();
     HistItem *item;
+    char n[sizeof("0")];
     char *p, *q;
 
+    sprintf(n, "%d", !zeroBasedLinkNo);
     /* FIXME: gettextize? */
     Strcat_charp(src, "<html>\n<head><title>History Page</title></head>\n");
     Strcat_charp(src, "<body>\n<h1>History Page</h1>\n<hr>\n");
-    Strcat_charp(src, "<ol>\n");
+    Strcat_m_charp(src, "<ol start=", n, ">\n", NULL);
     if (hist && hist->list) {
 	for (item = hist->list->last; item; item = item->prev) {
 	    q = html_quote(item->ptr);
